<html>
<head>
  <title>1）pVACseq</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="990"/>
<h1>1）pVACseq</h1>

<div>
<span><div><span style="font-size: 16pt; color: rgb(255, 0, 0);">1）</span><span style="box-sizing: border-box; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 16pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">pVACseq</span></div><div><span style="font-size: 14pt;">详细使用教程官网：</span><a href="https://pvac-seq.readthedocs.io/en/latest/prerequisites.html" style="font-size: 14pt;">https://pvac-seq.readthedocs.io/en/latest/prerequisites.html</a></div><div><span style="font-size: 14pt;"><span style="font-size: 14pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;"><img src="1）pVACseq_files/pvacseq-code-python.png" type="image/png" data-filename="pvacseq-code-python.png" width="1035"/></span></span></div><div><span style="font-size: 14pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">1.1. pVACseq 介绍</span></div><div><font style="font-size: 10pt;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">        1.1.1 软件下载</span></font></div><div>            1）文献地址：pVAC-Seq: A genome-guided in silico approach to identifying tumor neoantigens.    </div><div>            2）github地址：<a href="https://github.com/BioXiao/pVAC-Seq">https://github.com/BioXiao/pVAC-Seq</a></div><div>            3）依赖数据： DNA- and RNA-Seq</div><div>            4）安装： pip install pvacseq,更新： pip install pvacseq --upgrade</div><div>            5）理论基础：1）比较突变肽、野生肽与MHC的亲和力。</div><div>                             2）比较gene表达量。</div><div>            6）分析数据要求：1）非同义的体细胞突变，并且要注释上氨基酸变化和转录本信息。<span style="color: rgb(227, 0, 0); font-weight: bold;">（VEP注释）</span></div><div>                                    2）需要HLA分型结果。<span style="color: rgb(255, 0, 0);">（依据正常组织的数据）</span></div><div>                                    3）需要基因表达量结果。</div><div>            7）实现步骤：1） performing epitope prediction</div><div>                             2） integrating sequencing-based information</div><div>                             3） lastly, filtering neoantigen candidates</div><div><span style="font-size: 10pt;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">       1.1.2 数据准备：</span></span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">            1）annotating your vcf with VEP (</span><a href="https://blog.csdn.net/bio_meimei/article/details/89405740" style="font-size: 10pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;;">https://blog.csdn.net/bio_meimei/article/details/89405740</a><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">)</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">            2）adding co</span><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">verage data to your vcf (</span><a href="https://blog.csdn.net/bio_meimei/article/details/89948165" style="font-size: 10pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">https://blog.csdn.net/bio_meimei/article/details/89948165</a><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">) </span> <span style="font-size: 10pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">（</span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">bam-readcount</span><span style="font-size: 10pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">）</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">            3）adding expression data to your vcf (</span><a href="https://blog.csdn.net/bio_meimei/article/details/89949332" style="font-size: 10pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">https://blog.csdn.net/bio_meimei/article/details/89949332</a><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">)   </span><span style="font-size: 10pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">（</span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">cufflinks</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="font-size: 10pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">）</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">            4）creating a phased vcf of proximal variants(</span><a href="https://blog.csdn.net/bio_meimei/article/details/89949362" style="font-size: 10pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;;">https://blog.csdn.net/bio_meimei/article/details/89949362</a><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">)</span></div><div><br/></div><div><span style="min-height: 12pt;"><span style="font-size: 14pt; color: rgb(1, 1, 1); font-family: NimbusSanL-Bold; font-weight: bold;">1.2. Input File Preparation</span></span></div><div><span style="box-sizing: border-box; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><span style="font-size: 16pt;"> </span> <span style="font-size: 10pt;"> </span> <span style="font-size: 10pt; font-weight: bold;">1.2.1</span></span><span style="min-height: 12pt;"><font style="font-size: 10pt;"><span style="font-size: 10pt; color: rgb(32, 67, 92); font-weight: bold;">Running VEP</span></font></span></div><div><span style="min-height: 12pt;"><font style="font-size: 10pt;"><span style="font-size: 10pt; color: rgb(32, 67, 92); font-weight: bold;">            1）下载cache 文件：</span></font></span><a href="http://asia.ensembl.org/info/docs/tools/vep/script/vep_cache.html#cache" style="font-weight: bold;">http://asia.ensembl.org/info/docs/tools/vep/script/vep_cache.html#cache</a></div><div><span style="color: rgb(32, 67, 92);">             2）下载plugins：</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">git clone <a href="https://github.com/Ensembl/VEP_plugins.git" style="font-size: 9pt; font-family: NimbusMonL-Regu;">https://github.com/Ensembl/VEP_plugins.git</a></span></div><div>             3）非官方plugins下载，<span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">pvacseq install_vep_plugin &lt;VEP plugins directory&gt;</span></div><div><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">            Example VEP Command</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                ./vep \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                --input_file &lt;input VCF&gt; --output_file &lt;output VCF&gt; \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                --format vcf --vcf --symbol --terms SO --tsl\</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                --hgvs --fasta &lt;reference build fasta file location&gt; \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                --offline --cache [--dir_cache &lt;VEP cache directory&gt;] \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                --plugin Downstream --plugin Wildtype \</span></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 9pt; min-height: 11pt; font-family: NimbusMonL-Regu;">                [--dir_plugins &lt;VEP_plugins directory&gt;]</span> <span style="font-size: 9pt; color: rgb(227, 0, 0); font-family: NimbusMonL-Regu;">[--pick]</span> <span style="font-size: 9pt; min-height: 11pt; font-family: NimbusMonL-Regu;">[--transcript_version]</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">The --format vcf option specifies that the input file is in VCF format.</span></span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --vcf option will result in the output being written in VCF format.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --symbol option will include gene symbol in the annotation.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --terms SO option will result in Sequence Ontology terms being used for the consequences.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --tsl <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">option adds transcript support level information to the annotation.</span></span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --hgvs option will result in HGVS identifiers being added to the annotation.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • Using the --hgvs option requires the usage of the --fasta argument to specify the location of the reference genome build fasta file.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --offline option will eliminate all network connections for speed and/or privacy.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --cache option will result in the VEP cache being used for annotation.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --plugin Downstream option will run the Downstream plugin which will compute the downstream protein sequence after a frameshift.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --plugin Wildtype option will run the Wildtype plugin which will include the transcript protein sequence in the annotation.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --dir_cache &lt;VEP cache directory&gt; option may be needed if the VEP cache was downloaded to a different location than the default. The default location of the VEP cache is at $HOME/.vep.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --dir_plugins &lt;VEP_plugins directory&gt; option may need to be set depending on where the VEP_plugins were installed to.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --pick option might be useful</span> <span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">to limit the annotation to the “top” transcript</span> <span style="font-size: 9pt; font-family: NimbusMonL-Regu;">for each variant (the one for which the most dramatic consequence is predicted). Otherwise, VEP will annotate each variant with all possible transcripts. pVACseq will provide predictions for all transcripts in the VEP CSQ field. Running VEP without the --pick option can therefore drastically increase the runtime of pVACseq.</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">            • The --transcript_version option will</span> <span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">add the transcript version to the transcript identifiers.</span> <span style="font-size: 9pt; font-family: NimbusMonL-Regu;">This option might be needed if you intend to annotate your VCF with expression information. Particularly if your expression estimation tool uses versioned transcript identifiers (e.g. ENST00000256474.2).</span></div><div style="min-height: 11pt; text-align: left;"></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 10pt;">    </span><span style="min-height: 15pt; font-size: 10pt; font-weight: bold;">1.2.2 Adding coverage data to your VCF</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 15pt; font-size: 10pt; font-weight: bold;">            </span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">pip install vcf-annotation-tools. (</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">The</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">vcf-readcount-annotator</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">is part of the</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">vcf-annotation-tools</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">package.</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">)</span></div><div style="min-height: 11pt; text-align: left;">            1）<span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">Tumor DNA Coverage （</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">sample_name，</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">DP，AD，AF） (</span><span style="color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;"># acturally this step can be skip as the mutect2 provide the information of coverage</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">)</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">            2）</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">Tumor RNA Coverage </span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">（</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">sample_name，R</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">DP，RAD，RAF）</span></div><div style="min-height: 11pt; text-align: left;">            3）<span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">Normal DNA Coverage（</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">need to be a multi-sample (tumor/normal)</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">VCF，</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">sample_name，</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">--normal-sample-name</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">）</span></div><div style="min-height: 11pt; text-align: left;">             nocov.vcf--&gt;<span style="min-height: 12pt; font-size: 10pt; font-family: NimbusMonL-Regu;">vcf-readcount-annotator--&gt;</span><span style="min-height: 12pt; font-size: 10pt; font-family: NimbusRomNo9L-Regu;">bam-readcount --&gt;cov.vcf </span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 12pt; font-size: 10pt; font-family: NimbusRomNo9L-Regu;">            work:</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 12pt; font-size: 10pt; font-family: NimbusRomNo9L-Regu;">            1)</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">vt decompose -s &lt;input_vcf&gt; -o &lt;decomposed_vcf&gt;</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">              2)</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">bam-readcount -f &lt;reference_fasta&gt; -l &lt;site_list&gt; &lt;bam_file&gt; [-i] [-b 20]</span></div><div style="min-height: 11pt; text-align: left;">             3)<span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">bam_readcount_helper.py</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">              4)</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">vcf-readcount-annotator</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">    </span></div><div style="text-align: left;"><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">                </span><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;"># </span><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">    gene_expn_file: &lt;genes.fpkm_tracking file from Cufflinks&gt;</span></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">                </span><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;"># </span><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">    transcript_expn_file: &lt;isoforms.fpkm_tracking file from Cufflinks&gt;</span></div><div><span style="min-height: 15pt;"><font style="font-size: 10pt;"><span style="font-size: 10pt; font-weight: bold;">    1.2.3 Adding expression data to your VCF</span></font></span></div><div>            1)<span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">Transcript Expression （</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">single-sample VCF or</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">sample_name，</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">TX</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">）</span></div><div><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Medi;">            2）</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">Gene Expression （</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">single-sample VCF or</span> <span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusMonL-Regu;">sample_name，GX</span><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">）</span></div><div><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">            work：</span></div><div><span style="min-height: 12pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: NimbusRomNo9L-Regu;">            1）</span><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">vcf-expression-annotator input_vcf expression_file </span><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">kallisto|stringtie|cufflinks|custom gene|transcript</span></div><div><span style="min-height: 11pt; font-size: 9pt;"><br/></span></div><div>                 #   normal_snvs_coverage_file: &lt;bam-readcount output file for normal BAM and snvs&gt;</div><div>                 #   normal_indels_coverage_file: &lt;bam-readcount output file for normal BAM and indels&gt;</div><div>                 #   tdna_snvs_coverage_file: &lt;bam-readcount output file for tumor DNA BAM and snvs&gt;</div><div>                 #   tdna_indels_coverage_file: &lt;bam-readcount output file for tumor DNA BAM and indels&gt;</div><div>                 #   trna_snvs_coverage_file: &lt;bam-readcount output file for tumor RNA BAM and snvs&gt;</div><div>                 #  trna_indels_coverage_file: &lt;bam-readcount output file for tumor RNA BAM and indels&gt;</div><div><span style="min-height: 15pt;"><font style="font-size: 10pt;"><span style="font-size: 10pt; font-weight: bold;">    1.2.4 Creating a phased VCF of proximal variants</span></font></span></div><div>            work：</div><div>            1）Create the reference dictionary：java -jar picard.jar CreateSequenceDictionary R=reference.fa O=reference.dict</div><div>            2）Combine somatic and germline variants using GATK’s CombineVariants</div><div>                    /usr/bin/java -Xmx16g -jar /opt/GenomeAnalysisTK.jar \</div><div>                    -T CombineVariants \</div><div>                    -R reference.fa \</div><div>                    --variant germline.vcf \</div><div>                    --variant somatic.vcf \</div><div>                    -o combined_somatic_plus_germline.vcf \</div><div>                    --assumeIdenticalSamples</div><div>              3）Sort combined VCF using Picard</div><div>                    /usr/bin/java -Xmx16g -jar /opt/picard/picard.jar SortVcf \</div><div>                    I=combined_somatic_plus_germline.vcf \</div><div>                    O=combined_somatic_plus_germline.sorted.vcf \</div><div>                    SEQUENCE_DICTIONARY=reference.dict            </div><div>                4）Phase variants using GATK’s ReadBackedPhasing</div><div>                    /usr/bin/java -Xmx16g -jar /opt/GenomeAnalysisTK.jar \</div><div>                    -T ReadBackedPhasing \</div><div>                    -R reference.fa \</div><div>                    -I tumor.bam \</div><div>                    --variant combined_somatic_plus_germline.sorted.vcf \</div><div>                    -L combined_somatic_plus_germline.sorted.vcf \</div><div>                    -o phased.vcf                </div><div>                5）bgzip and index the phased VCF</div><div>                    bgzip -c phased.vcf &gt; phased.vcf.gz</div><div>                    tabix -p vcf phased.vcf.gz</div><div>                 6）bgzip and index the input VCF</div><div>                    bgzip -c input.vcf &gt; input.vcf.gz</div><div style="min-height: 11pt; text-align: left;">                    tabix -p vcf input.vcf.gz            <span style="color: rgb(32, 67, 92); font-family: NimbusSanL-Bold; font-weight: bold;">        </span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 17pt; font-size: 14.5pt; font-family: NimbusSanL-Bold; font-weight: bold;">1.3 Getting Started</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 17pt;"><font style="font-size: 12pt;"><span style="font-size: 12pt;"><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">        1.3.1 pvac-seq run</span></span></font></span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    pvacseq run \</span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    &lt;example_data_dir&gt;/input.vcf \</span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    Test \</span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    HLA-A*02:01,HLA-B*35:01,DRB1*11:01 \</span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    MHCflurry MHCnuggetsI MHCnuggetsII NNalign NetMHC PickPocket SMM SMMPMBEC SMMalign \</span></div><div style="text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    &lt;output_dir&gt; \</span></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                    -e 8,9,10         </span><span style="font-family: NimbusSanL-Bold; font-weight: bold;">  </span><span style="color: rgb(32, 67, 92); font-family: NimbusSanL-Bold; font-weight: bold;">         </span></div><div style="min-height: 11pt; text-align: left;"><font style="font-size: 12pt;"><span style="min-height: 12pt; font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">        1.3.2 </span><span style="min-height: 15pt; font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">Running pVACseq using Docker</span></font></div><div style="min-height: 11pt; text-align: left;"><font style="font-size: 12pt;"><span style="min-height: 15pt; font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">                </span></font><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;"> dockers版本：</span><span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">griffithlab/pvactools   </span><span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">on DockerHub （</span><span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">includes installations of the IEDB class I and class II tools at</span> <span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">/opt/iedb</span><span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">）</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 12pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">                    </span><span style="font-family: &quot;Times New Roman&quot;; font-weight: bold;">Please note that our Docker container already includes installations of the IEDB class I and class II tools at /opt/iedb (--iedb-install-directory /opt/iedb).</span></div><div style="min-height: 11pt; text-align: left;"><font face="Times New Roman" style="font-size: 10pt;"><span style="min-height: 15pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(32, 67, 92); font-weight: bold;">                    </span></font><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">docker run  \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                    -v /local/path/to/example_data_dir:/pvacseq_example_data \</span></div><div style="min-height: 11pt; text-align: left;"><span style="min-height: 11pt; font-size: 9pt; font-family: NimbusMonL-Regu;">                    -v /local/path/to/</span><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">output_dir:/pvacseq_output_data  \</span></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    -it griffithlab/pvactools</span></div><div style="min-height: 11pt; text-align: left;"><font style="font-size: 10pt;"><span style="font-size: 10pt; font-family: NimbusMonL-Regu; font-weight: bold;">                  work:</span></font></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    pvacseq run \</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    /pvacseq_example_data/input.vcf \        <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">&quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">input_file</span><span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    Test \                                                        <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">&quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">sample_name</span><span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    HLA-A*02:01,HLA-B*35:01,DRB1*11:01 \    <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(227, 0, 0);">&quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(227, 0, 0); font-family: NimbusMonL-Regu;">allele</span><span style="font-size: 9pt; color: rgb(227, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    MHCflurry MHCnuggetsI MHCnuggetsII NNalign NetMHC PickPocket SMM SMMPMBEC SMMalign \                                                                                                                                                                                                                    <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">&quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">[{MHCflurry,MHCnuggetsI,MHCnuggetsII,NNalign,NetMHC,NetMHCIIpan,</span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">NetMHCcons,NetMHCpan,PickPocket,SMM,SMMPMBEC,SMMalign} ...]</span><span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    /pvacseq_output_data \                                <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(227, 0, 0);">&quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(227, 0, 0); font-family: NimbusMonL-Regu;">output_dir</span><span style="font-size: 9pt; color: rgb(227, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    -e 8,9,10                                                    <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">    &quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">EPITOPE_LENGTH</span><span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">&quot; （8-11）</span></div><div style="text-align: left;"><span style="font-size: 9pt; font-family: NimbusMonL-Regu;">                    --iedb-install-directory /opt/iedb          <span style="font-size: 9pt; font-family: NimbusMonL-Regu; color: rgb(255, 0, 0);">  &quot;</span></span><span style="min-height: 11pt; font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">IEDB_INSTALL_DIRECTORY</span><span style="font-size: 9pt; color: rgb(255, 0, 0); font-family: NimbusMonL-Regu;">&quot;</span></div><div style="min-height: 11pt; text-align: left;"><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">    1.4 out files</span></font></div></div><div style="min-height: 11pt; text-align: left;"><span style="font-size: 9pt;">    </span><span style="font-size: 9pt;"><img src="1）pVACseq_files/Image.png" type="image/png" data-filename="Image.png" width="855"/></span></div><div><img src="1）pVACseq_files/Image [1].png" type="image/png" data-filename="Image.png" width="932"/></div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">    1.5 summary</span></font></div><div><br/></div><div><img src="1）pVACseq_files/Image [2].png" type="image/png" data-filename="Image.png" width="791"/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">    1.6）示例</span></font></div><div>            /mnt/X500/farmers/linhx/neoantigen/wes_rna/4.run_pvactool/seq_run.sh</div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">    1.7）question</span></font></div><div><br/></div><div>     1）发生了起始密码子突变的基因要不要考虑FS后产生新的ATG，导致新肽的产生？要不要考虑下游同时发生的移码突变？</div><div><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">        pVAC-Seq makes predictions for all transcripts of a variant that were annotated as</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="background-color: rgb(236, 240, 243); font-size: 0.95em; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; hyphens: none; color: rgb(0, 0, 0); font-variant-caps: normal; font-variant-ligatures: normal;">missense_variant</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">,</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="background-color: rgb(236, 240, 243); font-size: 0.95em; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; hyphens: none; color: rgb(0, 0, 0); font-variant-caps: normal; font-variant-ligatures: normal;">inframe_insertions</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">,</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="background-color: rgb(236, 240, 243); font-size: 0.95em; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; hyphens: none; color: rgb(0, 0, 0); font-variant-caps: normal; font-variant-ligatures: normal;">inframe_deletion</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">, or</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="background-color: rgb(236, 240, 243); font-size: 0.95em; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; hyphens: none; color: rgb(0, 0, 0); font-variant-caps: normal; font-variant-ligatures: normal;">frameshift_variant</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(0, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">by VEP as long as</span> <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(227, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">the transcript was not also annotated as</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(227, 0, 0); font-family: sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="background-color: rgb(236, 240, 243); font-size: 0.95em; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; hyphens: none; color: rgb(227, 0, 0); font-variant-caps: normal; font-variant-ligatures: normal;">start_lost。</span></div><div>     2）</div><div><br/></div><div><br/></div><div><br/></div><div><a href="1）pVACseq_files/PVAC-Seq.pdf.bak"><img src="1）pVACseq_files/0eed6a2b296a8f1765b96b541c5c737d.png" alt="PVAC-Seq.pdf.bak"></a></div><div><span style="font-size: 10pt;"><br/></span></div></span>
</div></body></html> 