<html>
<head>
  <title>gatk Mutect2:</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1429"/>
<h1>gatk Mutect2:</h1>

<div>
<span><div><span style="font-size: 16pt;">gatk Mutect2:</span></div><div>1) 教程地址：<a href="https://software.broadinstitute.org/gatk/documentation/article?id=24057">https://software.broadinstitute.org/gatk/documentation/article?id=24057</a></div><div>                     <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php">https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php</a></div><div>                    <a href="https://github.com/broadinstitute/gatk/blob/master/docs/mutect/mutect.pdf">https://github.com/broadinstitute/gatk/blob/master/docs/mutect/mutect.pdf</a></div><div>2）详细步骤：</div><div><br/></div><div><span style="font-size: 14pt; color: rgb(255, 0, 0);">A step-by-step guide to the new Mutect2 Read Orientation Artifacts Workflow</span></div><div><span style="font-weight: bold;">    1）First, run Mutect2 with the --f1r2-tar-gz argument.</span></div><div>       gatk <span style="font-weight: bold;">Mutect2 </span> \</div><div>       -R ref.fasta \</div><div>       -L intervals.interval_list \</div><div>       -I tumor.bam \</div><div>       -germline-resource af-only-gnomad.vcf \    ###下载地址：<a href="ftp://ftp.broadinstitute.org/bundle/Mutect2/">ftp://ftp.broadinstitute.org/bundle/Mutect2/</a></div><div>       -pon panel_of_normals.vcf   \</div><div>       --f1r2-tar-gz f1r2.tar.gz \</div><div>       -O unfiltered.vcf  \</div><div><span style="font-weight: bold;">        本地流程多出来的参数：</span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        --tmp-dir ./tmp</span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        --af-of-alleles-not-in-resource 0.0000025        ###</span><span style="color: rgb(77, 206, 29); font-weight: bold;">Population allele fraction assigned to alleles not found in germline resource.</span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        -A ClippingRankSumTest                                    #</span><span style="color: rgb(77, 206, 29); font-weight: bold;">One or more specific annotations to add to variant calls</span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        -bamout tumor_normal_m2.bam                       ##好像没啥用？</span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        </span><span style="color: rgb(77, 206, 29); font-weight: bold;">--genotype-germline-sites </span></div><div><span style="color: rgb(77, 206, 29); font-weight: bold;">        </span><span style="color: rgb(77, 206, 29); font-weight: bold;">-normal E11ND</span></div><div><span style="color: rgb(255, 0, 0);">        待选：</span></div><div><span style="color: rgb(255, 0, 0);">        --max-population-af,-max-af:Double            #</span><span style="color: rgb(255, 0, 0);">Maximum population allele frequency in tumor-only mode.  Default value: 0.01.</span></div><div><span style="color: rgb(255, 0, 0);">        </span><span style="color: rgb(255, 0, 0);">--max-reads-per-alignment-start:Integer        #</span><span style="color: rgb(255, 0, 0);">Maximum number of reads to retain per alignment start position. Reads above this threshold </span><span style="color: rgb(255, 0, 0);">will be downsampled. Set to 0 to disable.  Default value: 50.</span></div><div><span style="color: rgb(255, 0, 0);">        </span><span style="color: rgb(255, 0, 0);">--normal-lod:Double                         #</span><span style="color: rgb(255, 0, 0);">Log 10 odds threshold for calling normal variant non-germline.  Default value: 2.2.</span></div><div><span style="color: rgb(255, 0, 0);">        </span><span style="color: rgb(255, 0, 0);">--panel-of-normals,-pon:FeatureInput        ##</span><span style="color: rgb(255, 0, 0);">VCF file of sites observed in normal.  Default value: null.</span></div><div><span style="color: rgb(255, 0, 0);">        </span><span style="color: rgb(255, 0, 0);">--tumor-lod-to-emit,-emit-lod:Double        ##</span><span style="color: rgb(255, 0, 0);">Log 10 odds threshold to emit variant to VCF.  Default value: 3.0.</span></div><div><span style="color: rgb(255, 0, 0);">        </span><span style="color: rgb(255, 0, 0);">--max-mnp-distance,-mnp-dist:Integer        #</span><span style="color: rgb(255, 0, 0);">Two or more phased substitutions separated by this distance or less are merged into MNPs。</span><span style="color: rgb(255, 0, 0);">Default value: 1.</span></div><div><span style="color: rgb(255, 0, 0);">        </span></div><div><span style="font-weight: bold;">    2）Next, pass this raw data to LearnReadOrientationModel:</span></div><div>        gatk <span style="font-weight: bold;">LearnReadOrientationModel</span>  \</div><div>            -I f1r2.tar.gz  \</div><div>            -O read-orientation-model.tar.gz</div><div><span style="font-weight: bold;">    3）Run GetPileupSummaries to summarize read support for a set number of known variant sites.</span></div><div>        gatk <span style="color: rgb(255, 70, 53); font-weight: bold;">GetPileupSummaries</span> \</div><div>        -I tumor.bam \                                                             #BAM/SAM/CRAM file containing reads</div><div>        -V chr17_small_exac_common_3_grch38.vcf.gz \        #A VCF file containing variants and allele frequencies  Required.</div><div>        -L chr17_small_exac_common_3_grch38.vcf.gz \         #One or more genomic intervals over which to operate</div><div>        -O getpileupsummaries.table                                       #The output table  Required</div><div>        <span style="color: rgb(255, 70, 53); font-weight: bold;">待选：</span></div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">        --maximum-population-allele-frequency,-max-af:Double        #</span><span style="color: rgb(255, 70, 53); font-weight: bold;">Maximum population allele frequency of sites to consider.  Default value: 0.2.</span></div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">        </span><span style="color: rgb(255, 70, 53); font-weight: bold;">--minimum-population-allele-frequency,-min-af:Double         #</span><span style="color: rgb(255, 70, 53); font-weight: bold;">Minimum population allele frequency of sites to consider.  A low value increases accuracy at the expense of speed.  Default value: 0.01.</span></div><div> <span style="font-weight: bold;">   4）Estimate contamination with CalculateContamination.</span></div><div>        gatk <span style="color: rgb(255, 0, 0); font-weight: bold;">CalculateContamination</span> \</div><div>        -I getpileupsummaries.table \</div><div>        -tumor-segmentation segments.table \        ###The <span style="font-weight: bold;">output table</span> containing segmentation of the tumor by minor allele fraction</div><div>        -O calculatecontamination.table \</div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">         待选：</span></div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">        --matched-normal normal.table</span></div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">        </span><span style="color: rgb(255, 70, 53); font-weight: bold;">--high-coverage-ratio-threshold:Double        </span><span style="color: rgb(255, 70, 53); font-weight: bold;">The maximum coverage relative to the mean.  Default value: 3.0.</span></div><div><span style="color: rgb(255, 70, 53); font-weight: bold;">        </span><span style="color: rgb(255, 70, 53); font-weight: bold;">--low-coverage-ratio-threshold:Double        </span><span style="color: rgb(255, 70, 53); font-weight: bold;">The minimum coverage relative to the median.  Default value: 0.5.</span></div><div><br/></div><div>   <span style="font-weight: bold;"> 5）pass the learned read orientation model to FilterMutectCallswith the -ob-priors argument:</span></div><div>        gatk <span style="font-weight: bold;">FilterMutectCalls </span></div><div>        -V unfiltered.vcf \</div><div>       --tumor-segmentation segments.table \</div><div>       --contamination-table contamination.table \</div><div>       --ob-priors read-orientation-model.tar.gz \</div><div>       -O filtered.vcf    \</div><div>        -R tgp_phase2_flat/hs37d5.fa </div><div>        <span style="font-weight: bold;">本地流程多出来的参数：</span></div><div><span style="color: rgb(107, 231, 72); font-weight: bold;">        -</span><span style="color: rgb(107, 231, 72); font-weight: bold;">-stats</span> <span style="color: rgb(107, 231, 72); font-weight: bold;">unfiltered.vcf</span><span style="color: rgb(107, 231, 72); font-weight: bold;">.stats</span><span style="color: rgb(107, 231, 72); font-weight: bold;">   \               </span> <span style="color: rgb(107, 231, 72); font-weight: bold;">###输入</span><span style="font-weight: bold;">Mutect2  步骤的附带输出结果，如果包含多个stats文件，还需要合并一下才行。</span></div><div><span style="font-weight: bold;">                                                                        </span><span style="box-sizing: border-box; overflow: auto; font-size: 14px; word-break: break-all; word-wrap: break-word; background-color: rgb(245, 245, 245); border: 1px solid rgb(204, 204, 204); border-radius: 0px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: pre-wrap; color: rgb(51, 51, 51); font-family: Consolas, Monaco, &quot;Lucida Console&quot;, &quot;Courier New&quot;, monospace; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold; line-height: 1.4;">gatk MergeMutectStats -stats unfiltered1.vcf.stats -stats unfiltered2.vcf.stats -O merged.stats</span></div><div><span style="color: rgb(107, 231, 72); font-weight: bold;">        --filtering-stats FMC.filtering.stats \</span></div><div><br/></div><div><br/></div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; color: rgb(255, 0, 0); font-weight: bold;">A step-by-step guide to the new Mutect2 Panel of Normals Workflow</span></font></div><div>    The three steps to create a panel of normals are:</div><div>  <span style="font-weight: bold;">  Step 1: Run Mutect2 in tumor-only mode for each normal sample:</span></div><div>        gatk Mutect2 -R reference.fasta -I normal1.bam   <span style="font-weight: bold;">--max-mnp-distance 0</span> -O normal1.vcf.gz</div><div>        gatk Mutect2 -R reference.fasta -I normal2.bam   --max-mnp-distance 0 -O normal2.vcf.gz</div><div><span style="font-weight: bold;">    Step 2: Create a GenomicsDB from the normal Mutect2 calls:</span></div><div>        gatk GenomicsDBImport -R reference.fasta -L intervals.interval_list \</div><div>            --genomicsdb-workspace-path pon_db \</div><div>            -V normal1.vcf.gz \</div><div>            -V normal2.vcf.gz \</div><div>            -V normal3.vcf.gz</div><div><span style="font-weight: bold;">    Step 3: Combine the normal calls using CreateSomaticPanelOfNormals:</span></div><div>        gatk CreateSomaticPanelOfNormals -R reference.fasta \</div><div>          --germline-resource af-only-gnomad.vcf.gz \</div><div>          -V gendb://pon_db \</div><div>          -O pon.vcf.gz</div><div>    In the third step, the AF-only gnomAD resource is the same public GATK resource used by Mutect2 (when calling tumor samples, but not in Step 1, above).    </div><div><br/></div><div><br/></div><div><br/></div><div>   </div></span>
</div></body></html> 