<html>
<head>
  <title>GATK命令大全：</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1275"/>
<h1>GATK命令大全：</h1>

<div>
<span><div><span style="font-size: 16pt; color: rgb(255, 0, 0);">GATK命令大全：</span></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Diagnostics and Quality Control Tools</span></font></div><div>        1. ASEReadCounter    -&gt;     计算特异表达的等位基因的reads数</div><div>        2. AnalyzeCovariates    -&gt;     Create plots to visualize base recalibration results</div><div><span style="color: rgb(255, 0, 0);">        3. CallableLoci    -&gt;    统计基因组中可call突变，比对质量以及其他区域</span></div><div>            REF_N: 基因组N区</div><div>            PASS: 可call突变</div><div>            NO_COVERAGE: 未覆盖区</div><div>            LOW_COVERAGE: 低于最低深度的区域</div><div>            EXCESSIVE_COVERAGE: 高于最高深度的区域</div><div>            POOR_MAPPING_QUALITY: 低比对质量区域</div><div>        4. CheckPileup    -&gt;    比较GATK pileup 与Samtools 的pileup格式的文件</div><div>        5. CompareCallableLoci    -&gt;    不同软件得到的CallableLoci文件之间的比较</div><div><span style="color: rgb(255, 0, 0);">        6. ContEst    -&gt;    估计样本交叉污染</span></div><div>        7. CountBases    -&gt;    统计bam文件中的碱基数</div><div>        8. CountIntervals    -&gt;    合并连续区域(ROD格式的文件)</div><div>        9. CountLoci    -&gt;    计算位点数</div><div>        10. CountMales    -&gt;    计算男性样本的reads数</div><div>        11. CountRODs    -&gt;    Count the number of ROD objects encountered</div><div>        12. CountRODsByRef    -&gt;    Count the number of ROD objects encountered along the reference</div><div>        13. CountReadEvents    -&gt;    Count the number of read events</div><div>        14. CountReads    -&gt;    统计reads数</div><div>        15. CountTerminusEvent    -&gt;    统计末端碱基未匹配的reads数</div><div><span style="color: rgb(255, 0, 0);">        16. DepthOfCoverage    -&gt;    统计覆盖度</span></div><div>        17. DiagnoseTargets    -&gt;    Analyze coverage distribution and validate read mates per interval and per sample</div><div>        18. DiffObjects    -&gt;    两个或多个bam/vcf文件之间的比较</div><div>        19. ErrorRatePerCycle    -&gt;    计算read的错误率</div><div><span style="color: rgb(255, 0, 0);">        20.FastaStats    -&gt;    统计参考基因组碱基信息</span></div><div><span style="color: rgb(255, 0, 0);">        21. FindCoveredIntervals    -&gt;    输出覆盖度高于指定值的区间</span></div><div>        22. FlagStat    -&gt;    与 samtools flagstat 功能相同</div><div>        23. GCContentByInterval    -&gt;    统计GC含量</div><div>        24. GatherBqsrReports    -&gt;    </div><div>        25. Pileup    -&gt;    与samtools pileup 功能相同</div><div>        26. PrintRODs    -&gt;    Print out all of the RODs in the input data set</div><div><span style="color: rgb(255, 0, 0);">        27. QualifyMissingIntervals    -&gt;    统计以下质量指标</span></div><div>            Average Base Quality</div><div>            Average Mapping Quality</div><div>            Average Depth</div><div>            GC Content</div><div>            Position in the target (Integer.MIN_VALUE if no overlap)</div><div>            Length of the overlapping target (zero if no overlap)</div><div>            Coding Sequence / Intron (optional)</div><div>            Length of the uncovered interval</div><div>        28. ReadClippingStats    -&gt;    stat read length, number of clipping events, and length of the clipping</div><div>        29. ReadGroupProperties    -&gt;    统计基因组和测序信息</div><div>        30. ReadLengthDistribution    -&gt;    统计read长度信息</div><div>        31. SimulateReadsForVariants    -&gt;    模拟生成vcf文件的bam</div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Sequence Data Processing Tools</span></font></div><div><span style="color: rgb(255, 0, 0);">        1. BaseRecalibrator    -&gt;    碱基质量重新校准</span></div><div>        2. ClipReads    -&gt;    可以根据质量值，位置，比对信息来剪切reads</div><div>        3. IndelRealigner    -&gt;    对已知的INDEL区域的reads重新比对，尽量减少不匹配的reads</div><div>        4.LeftAlignIndels    -&gt;    左对齐bam文件 INDLE中的reads</div><div>        5. PrintReads    -&gt;    可用来合并bam</div><div>        6. RealignerTargetCreator    -&gt;    确定要进行重新比对的区域</div><div>        7. SplitNCigarReads    -&gt;    用于RNA-seq将落在内含子区间的 reads 片段直接切除，并对 MAPQ 进行调整</div><div><span style="color: rgb(255, 0, 0);">        8.SplitSamFile    -&gt;    按照样本拆分bam文件</span></div><div><font style="font-size: 14pt;"><br/></font></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Variant Discovery Tools</span></font></div><div>        1.ApplyRecalibration    -&gt;    VQSR 第一阶段</div><div>        2. CalculateGenotypePosteriors    -&gt;    Calculate genotype posterior likelihoods given panel data</div><div>        3. GATKPaperGenotyper    -&gt;    Simple Bayesian genotyper used in the original GATK paper</div><div><span style="color: rgb(255, 0, 0);">        4. GenotypeGVCFs    -&gt;    将多个GVCF文件合并为VCF</span></div><div><span style="color: rgb(255, 0, 0);">        5. HaplotypeCaller    -&gt;    加入了实时de-novo 算法来Call SNPs 和 indels</span></div><div><span style="color: rgb(255, 0, 0);">        6. MuTect2    -&gt;    用于肿瘤细胞call变异</span></div><div>        7. RegenotypeVariants    -&gt;    根据 VCF 文件中的 PLs 或者 GLs重新call 突变</div><div>        8. UnifiedGenotyper    -&gt;    以前call突变的方式</div><div>        9. VariantRecalibrator    -&gt;    VQSR 第二阶段</div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Variant Evaluation Tools</span></font></div><div>        1. GenotypeConcordance    -&gt;    比较两个突变文件基因型的一致性</div><div>        2. ValidateVariants    -&gt;    VCF文件信息验证</div><div>        3. VariantEval    -&gt;    突变评估,统计一些质量指标，比如：过滤掉的snp比例,转换突变与颠换的比率，以及突变信息的聚类等</div><div><span style="color: rgb(255, 0, 0);">        4. VariantFiltration    -&gt;    根据 vcf文件INFO 和 FORMAT过滤文件</span></div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Variant Manipulation Tools</span></font></div><div><span style="color: rgb(255, 0, 0);">        1. CatVariants    -&gt;    合并VCF文件，连接多个样本的非重叠区</span></div><div><span style="color: rgb(255, 0, 0);">        2. CombineGVCFs    -&gt;    合并GVCF文件</span></div><div>        3. CombineVariants    -&gt;    ROD格式的突变文件的合并</div><div>        4. HaplotypeResolver    -&gt;    Haplotype-based resolution of variants in separate callsets</div><div><span style="color: rgb(255, 0, 0);">        5. LeftAlignAndTrimVariants    -&gt;    对vcf文件的indel修剪到最短</span></div><div>        6. PhaseByTransmission    -&gt;    Compute the most likely genotype combination and phasing for trios and parent/child pairs</div><div><span style="color: rgb(255, 0, 0);">        7. RandomlySplitVariants    -&gt;    随机拆分VCF文件</span></div><div>        8. ReadBackedPhasing    -&gt;    获取突变的物理位置(获取突变到底是在哪条reads上?)</div><div>        9. SelectHeaders    -&gt;    获取头文件信息</div><div><span style="color: rgb(255, 0, 0);">        10. SelectVariants    -&gt;    根据条件对vcf进行筛选</span></div><div>        11. ValidationSiteSelector    -&gt;    Randomly select variant records according to specified options</div><div><span style="color: rgb(255, 0, 0);">        12. VariantAnnotator    -&gt;    根据前后关系进行注释？详见Annotation Modules</span></div><div><span style="color: rgb(255, 0, 0);">        13. VariantsToAllelicPrimitives    -&gt;    delins拆分为多突变</span></div><div>        14. VariantsToBinaryPed    -&gt;    把VCF文件转换成二进制文件</div><div>        15. VariantsToTable    -&gt;    从VCF文件里提取特定的字段(比如：CHROM,POS,ID,QUAL,AC)</div><div><span style="color: rgb(255, 70, 53);">        16. VariantsToVCF    -&gt;    将其他突变文件转变为VCf格式</span></div><div><br/></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">Useful Tools:</span></font></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        1）</span><span style="font-size: 11pt; color: rgb(77, 206, 29); font-family: &quot;Times New Roman&quot;; font-weight: bold;"> gatk ScatterIntervalsByNs  -&gt;   </span> <span style="font-size: 11pt;"><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;"> 计算基因组已知和未知区域  </span></span> <span style="font-size: 11pt; color: rgb(77, 206, 29); font-family: &quot;Times New Roman&quot;; font-weight: bold;"> --OUTPUT hs37d5.fa_nonN_region --REFERENCE hs37d5.fa</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        2） gatk SamToFastq --FASTQ  xx_1.fq --SECOND_END_FASTQ  xx_2.fq</span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">--UNPAIRED_FASTQ xx_unpair.fq </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">--INPUT xx.bam </span></div><div><span style="font-size: 11pt;"><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">    </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">    </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">    </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">    （非配对的bam会报错</span></span> <span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">Failed to detect whether we are running on Google Compute Engine.</span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">）</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">             </span> <span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">  bam提取</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">：</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">samtools view recaled.bam 1:1-1000000 -o tmp.bam <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;; color: rgb(255, 0, 0); font-weight: bold;">（这个命令好像只能处理经过read-name排序的样本，根据位置排过序的好像搞出来有问题。）</span></span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">               </span> <span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">bam2fq</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">：</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">samtools fastq -1 xx_1.fq -2 xx_2.fq -sxx_unpair.fq  E11ND_chr1.bam</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        3)    gatk SortSam -I hlas.bam -O hlas.csorted.bam -SO coordinate  <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;; color: rgb(255, 0, 0); font-weight: bold;">(根据基因组位置排序)</span></span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        4)   </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">gatk FilterSamReads -I allele.bam</span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">-FILTER includeReadList -READ_LIST_FILE readName.txt</span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">-O allele.filtered.bam</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        5)   </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: pre-wrap; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 11pt; color: rgb(25, 31, 37); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">gatk LeftAlignAndTrimVariants --reference hs37d5.fa --variant decompose.vcf --output leftTrim.vcf</span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: pre-wrap; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 11pt;"><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">（INDEL 对齐）</span></span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">     </span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">  6)   </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">gatk AddOrReplaceReadGroups -I input.bam -O out.bam -ID 199003847TD_LB1_PU1 -SM 199003847TD -PU PU1 -LB LB1 -PL ILLUMINA </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">（</span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">ReadGroups替换</span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">）</span></div><div><span style="font-size: 11pt;"><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        7)   </span></span></div><div><span style="font-size: 11pt;"><br/></span></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div></span>
</div></body></html> 