<html>
<head>
  <title>gatk 【WGS、WES】somatic-CNA 分析流程：</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1273"/>
<h1>gatk 【WGS、WES】somatic-CNA 分析流程：</h1>

<div>
<span><div><span style="font-size: 16pt; color: rgb(255, 0, 0);">gatk 【WGS、WES】somatic-CNA 分析流程：</span></div><div><br/></div><div><span style="font-size: 14pt;">1）docker 镜像</span>：<a href="https://hub.docker.com/r/broadinstitute/gatk/">https://hub.docker.com/r/broadinstitute/gatk/</a>  （docker pull broadinstitute/gatk）</div><div><span style="font-size: 14pt;">2）教程地址</span><span style="font-size: 14pt; font-weight: bold;">：</span><a href="https://software.broadinstitute.org/gatk/documentation/topic?name=tutorials">https://software.broadinstitute.org/gatk/documentation/topic?name=tutorials</a></div><div><span style="font-size: 14pt;">3）分析对象：</span> <span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">The same workflow steps apply to targeted exome and whole genome sequencing data.</span></div><div><span style="font-size: 14pt; color: rgb(255, 0, 0);">4）本地流程：</span><span style="font-size: 14pt; color: rgb(255, 0, 0);">/mnt/X500/farmers/luozw/factory/20190821_15WGS_sv_cnv/cnv_result/gatk_somatic_cnv/get_gatk_somatic_cnv.sh</span></div><div><span style="font-size: 14pt;"><br/></span></div><div><span style="font-size: 14pt;">主要步骤</span></div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">1. 生成interval bed文件，并生成样本interval reads统计结果。</span></font></div><div>    1.1） PreprocessIntervals</div><div>        gatk PreprocessIntervals -R hg19_ref/hs37d5.fa --bin-length 50000 --interval-merging-rule OVERLAPPING_ONLY -O hs37d5.fa.interval_list</div><div>        gatk PreprocessIntervals \</div><div>        -L targets_C.interval_list \        #    The -L argument is optional.</div><div>        -R /gatk/ref/Homo_sapiens_assembly38.fasta \    #reference</div><div>        --bin-length 0 \            #50000,default 1000 for whole genome or 0 for exomes.</div><div>        --interval-merging-rule OVERLAPPING_ONLY \        #to prevent the tool from merging abutting intervals. [4]</div><div>        -O sandbox/targets_C.preprocessed.interval_list        #out_file</div><div>    1.2）CollectReadCounts</div><div>        gatk CollectReadCounts -I recaled.bam -L hs37d5.fa.interval_list --interval-merging-rule OVERLAPPING_ONLY -O recaled_tumor.counts.hdf5</div><div>        gatk CollectReadCounts \</div><div>        -I tumor.bam \</div><div>        -L targets_C.preprocessed.interval_list \</div><div>        --interval-merging-rule OVERLAPPING_ONLY \</div><div>        -O sandbox/tumor.counts.hdf5</div><div><span style="color: rgb(255, 0, 0);">        ###已测试，运行流畅。</span></div><div><br/></div><div><span style="font-size: 12pt; font-weight: bold;">2. 根据样本interval reads生成PON文件。</span></div><div>        Generate a CNV panel of normals with CreateReadCountPanelOfNormals     </div><div>        gatk --java-options &quot;-Xmx6500m&quot; CreateReadCountPanelOfNormals \</div><div>        -I HG00133.alt_bwamem_GRCh38DH.20150826.GBR.exome.counts.hdf5 \</div><div>        -I HG00733.alt_bwamem_GRCh38DH.20150826.PUR.exome.counts.hdf5 \</div><div>        -I NA19654.alt_bwamem_GRCh38DH.20150826.MXL.exome.counts.hdf5 \</div><div><span style="color: rgb(255, 0, 0); font-weight: bold;">        --annotated-intervals:File \  </span> <span style="font-size: medium; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: 微软雅黑; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="font-size: medium; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: 微软雅黑; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">annotated_intervals_GC 文件，这里是作为可选非必要参数。</span></div><div>        --minimum-interval-median-percentile 5.0 \</div><div>        -O sandbox/cnvponC.pon.hdf5</div><div>        ###可以接受对照的一个样本,但是效果没有PON对照分析结果好。The tool will accept a single sample, e.g. the matched-normal.</div><div><span style="color: rgb(255, 0, 0);">        ###已测试，运行流畅。</span></div><div><br/></div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">3. 根据PON对分析数据进行降噪处理</span></font></div><div>        Standardize and denoise case read counts against the PoN with DenoiseReadCounts</div><div>        gatk --java-options &quot;-Xmx12g&quot; DenoiseReadCounts \</div><div>        -I hcc1143_T_clean.counts.hdf5 \</div><div>        --count-panel-of-normals cnvponC.pon.hdf5 \</div><div><span style="color: rgb(255, 0, 0); font-weight: bold;">        --annotated-intervals:File \  </span> <span style="font-size: medium; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: 微软雅黑; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="font-size: medium; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: 微软雅黑; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">annotated_intervals_GC 文件，这里是作为可选非必要参数。</span></div><div>        --standardized-copy-ratios sandbox/hcc1143_T_clean.standardizedCR.tsv \   ##输出文件</div><div>        --denoised-copy-ratios sandbox/hcc1143_T_clean.denoisedCR.tsv              ##输出文件</div><div>        <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。</span></div><div><span style="color: rgb(255, 0, 0);">        In the first transformation, the tool standardizes counts by the PoN median counts. The standarization includes log2 transformation and normalizing the counts data to center around one.</span></div><div><span style="color: rgb(255, 0, 0);">        In the second transformation, the tool denoises the standardized copy ratios using the principal components of the PoN.</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">        </span> <span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">The input counts are then standardized by：</span></div><div><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">1) transforming to fractional coverage, </span></div><div><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">2) performing optional explicit GC-bias correction (if the panel contains GC-content annotated intervals), </span></div><div><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">3) filtering intervals to those contained in the panel, </span></div><div><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">4) dividing by interval medians contained in the panel, </span></div><div><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">5) dividing by the sample median, and </span></div><div><span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">    </span><span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">6) transforming to log2 copy ratio.</span><span style="font-size: 10pt; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(34, 34, 34); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span></div><div><br/></div><div><span style="font-size: 12pt; font-weight: bold;">4. 对降噪处理的数据作图，直观评估降噪效果</span></div><div>       Plot standardized and denoised copy ratios with PlotDenoisedCopyRatios    </div><div>        gatk PlotDenoisedCopyRatios \</div><div>        --standardized-copy-ratios hcc1143_T_clean.standardizedCR.tsv \</div><div>        --denoised-copy-ratios hcc1143_T_clean.denoisedCR.tsv \</div><div>        --sequence-dictionary hg19_ref/hs37d5.dict \</div><div>        --minimum-contig-length 1000000    \    #  46709983 \</div><div>        --output sandbox/plots \</div><div>        --output-prefix hcc1143_T_clean</div><div>          <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。</span></div><div><br/></div><div>        This produces six files in the plots directory--two PNG images and four text files as follows.</div><div>        hcc1143_T_clean.denoised.png        ##降噪结果图</div><div>        hcc1143_T_clean.denoisedLimit4.png    ##降噪结果图</div><div>        hcc1143_T_clean.standardizedMAD.txt     ##gives the MAD for standardized copy ratios.</div><div>        hcc1143_T_clean.denoisedMAD.txt         ##gives the MAD for denoised copy ratios.</div><div>        hcc1143_T_clean.deltaMAD.txt             ##gives the difference between standardized MAD and denoised MAD.</div><div>        hcc1143_T_clean.scaledDeltaMAD.txt         ##gives the fractional difference (standardized MAD - denoised MAD)/(standardized MAD).</div><div>    </div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">5. 分析tumor bam 和normal bam 等位基因变异结果</span></font></div><div>        Count ref and alt alleles at common germline variant sites using CollectAllelicCounts</div><div>        5.1）Collect counts at germline variant sites for the matched-control</div><div>            gatk --java-options &quot;-Xmx3g&quot; CollectAllelicCounts \</div><div>            -L chr17_theta_snps.interval_list \</div><div>            -I normal.bam \</div><div>            -R /gatk/ref/Homo_sapiens_assembly38.fasta \</div><div>            -O sandbox/hcc1143_N_clean.allelicCounts.tsv</div><div>        5.2）Collect counts at the same sites for the case sample</div><div>            gatk --java-options &quot;-Xmx3g&quot; CollectAllelicCounts \</div><div>            -L chr17_theta_snps.interval_list \</div><div>            -I tumor.bam \</div><div>            -R /gatk/ref/Homo_sapiens_assembly38.fasta \</div><div>            -O sandbox/hcc1143_T_clean.allelicCounts.tsv    </div><div>            </div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">6. 根据copy ratios结果和等位基因变异结果分析可靠的CNV，并连接成片段</span></font></div><div>    Group contiguous copy ratios into segments with ModelSegments</div><div>    6.1）gatk --java-options &quot;-Xmx4g&quot; ModelSegments \</div><div>        --denoised-copy-ratios hcc1143_T_clean.denoisedCR.tsv \</div><div>        --allelic-counts hcc1143_T_clean.allelicCounts.tsv \</div><div>        --normal-allelic-counts hcc1143_N_clean.allelicCounts.tsv \</div><div>        --output sandbox \</div><div>        --output-prefix hcc1143_T_clean</div><div>    <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。</span></div><div>    6.2）This produces nine files each with the basename hcc1143_T_clean in the current directory and listed below.</div><div>        hcc1143_T_clean.modelBegin.seg        #Fits initial model. Writes initial segments to modelBegin.seg</div><div>        hcc1143_T_clean.modelFinal.seg        #final segments</div><div>        hcc1143_T_clean.cr.seg                #final copy-ratio segments to</div><div>        hcc1143_T_clean.modelBegin.af.param        #allele-fraction global parameters</div><div>        hcc1143_T_clean.modelBegin.cr.param        #posterior summaries for copy-ratio global parameter</div><div>        hcc1143_T_clean.modelFinal.af.param        #posterior summaries for allele-fraction global parameters</div><div>        hcc1143_T_clean.modelFinal.cr.param        #posterior summaries for copy-ratio global parameters</div><div>        hcc1143_T_clean.hets.normal.tsv        #Allelic counts for sites in the control that are heterozygous are written to hets.normal.tsv.</div><div>        hcc1143_T_clean.hets.tsv            #Allelic counts for sites in the tumor that are heterozygous are written to hets.tsv.</div><div>        </div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">7. 生成CNV最终结果</span></font></div><div>    Call copy-neutral, amplified and deleted segments with CallCopyRatioSegments</div><div>        gatk CallCopyRatioSegments \</div><div>        --input hcc1143_T_clean.cr.seg \</div><div>        --output sandbox/hcc1143_T_clean.called.seg    </div><div>     <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。</span></div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">8. 结果作图</span></font></div><div>    Plot modeled copy ratio and allelic fraction segments with PlotModeledSegments</div><div>        gatk PlotModeledSegments \</div><div>        --denoised-copy-ratios hcc1143_T_clean.denoisedCR.tsv \</div><div>        --allelic-counts hcc1143_T_clean.hets.tsv \</div><div>        --segments hcc1143_T_clean.modelFinal.seg \</div><div>        --sequence-dictionary Homo_sapiens_assembly38.dict \</div><div>        --minimum-contig-length 1000000 \        #46709983 \</div><div>        --output sandbox/plots \</div><div>        --output-prefix hcc1143_T_clean        </div><div>     <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。</span></div><div><br/></div><div><font style="font-size: 14pt;"><br/></font></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; color: rgb(255, 0, 0); font-weight: bold;">4） GC矫正（在此工具中作为可选部分，不做矫正也能分析出结果）</span></font></div><div>        4.1）生成GC文件</div><div>             gatk AnnotateIntervals \</div><div>                  -R reference.fa \</div><div>                  -L intervals.interval_list \</div><div>                  --interval-merging-rule OVERLAPPING_ONLY \</div><div>                  -O annotated_intervals.tsv</div><div>         <span style="color: rgb(255, 0, 0);">###已测试，运行流畅。通过比较发现GC矫正 对结果影响不大。可能因为PON矫正已经达到不错的效果。</span></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div>    </div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">            </span></div><div><br/></div></span>
</div></body></html> 