<html>
<head>
  <title>NGS 常用分析软件</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="554"/>
<h1>NGS 常用分析软件</h1>

<div>
<span><div><div style="margin: 0mm 0mm 5.28mm; text-indent: 0mm; padding-top: 0mm; padding-bottom: 0mm; min-height: 29pt;"><span style="text-indent: 0mm; min-height: 29pt; font-family: ����;"><font style="font-size: 16pt; color: rgb(255, 0, 0);">NGS 常用分析软件</font></span></div><div style="margin: 0mm 0mm 5.28mm; text-indent: 0mm; padding-top: 0mm; padding-bottom: 0mm; min-height: 29pt;"><br/></div><div style="margin: 0mm 0mm 5.28mm; text-indent: 0mm; padding-top: 0mm; padding-bottom: 0mm; min-height: 29pt;"><a href="NGS 常用分析软件_files/第五节：常用生物信息工具-20150323.pdf"><img src="NGS 常用分析软件_files/68963e39252fb283a34f38d7956c0349.png" alt="第五节：常用生物信息工具-20150323.pdf"></a></div><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">1. Mapping</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://sourceforge.net/apps/mediawiki/bfast/index.php?title=Main_Page%20\t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">BFAST</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A fast and accurate tool for mapping of short reads to reference sequences.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://bio-bwa.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">BWA</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A fast light-weighted tool that aligns short nucleotide sequences to a sequence database.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://bowtie-bio.sourceforge.net/index.shtml%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Bowtie</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An ultrafast, memory-efficient short read aligner.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://bioinfo.cgrb.oregonstate.edu/docs/solexa/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">ELAND</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A very fast alignment algorithms from Illumina company.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://maq.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">MAQ</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software that builds mapping assemblies from short reads generated by the next-generation sequencing machines.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://compbio.cs.toronto.edu/shrimp/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SHRiMP</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for aligning genomic reads against a target genome.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://soap.genomics.org.cn/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SOAP</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool package that provides full solution to next generation sequencing data analysis (including a alignment tool SOAPaligner/soap2 etc).</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://solid.community.appliedbiosystems.com/groups/solid-news/blog/2010/11/10/solid-bioscope-software-13-now-available%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SOLiD bioscope</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package that is designed specifically to optimize the accuracy of the ABI SOLiD colorspace data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://bibiserv.techfak.uni-bielefeld.de/swift/welcome.html%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SWIFT</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software collection for fast index-based sequence comparison.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://tophat.cbcb.umd.edu/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">TopHat</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A spliced read mapper for RNA-Seq.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">2. SNV Detection</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.illumina.com/software/genome_analyzer_software.ilmn%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CASAVA</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: The internal assembler and variant caller Illumina company utilized.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.broadinstitute.org/gsa/wiki/index.php/Unified_genotyper%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">GATK</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A multiple-sample, technology-aware SNV and indel caller.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://compbio.bccrc.ca/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">JointSNVMix</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A probabilistic model for detection of somatic mutations in normal/tumour pair.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://samtools.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SAMtools</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A set of utilities for post-processing alignments in the SAM format, such as indexing, variant caller and alignment viewer.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://compbio.bccrc.ca/?page_id=204%20\t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SNVMix</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for SNV calling based on probabilistic binomial mixture model.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://soap.genomics.org.cn/soapsnp.html%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SOAPsnp</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for identifying SNVs by Beijing Genomics Institute (BGI).</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="ftp://strelka@ftp.illumina.com/%20%5Ct%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Strelka</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for somatic small-variant calling from sequenced tumor-normal sample pairs.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.wustl.edu/software/somaticsniper%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SomaticSniper</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A program to identify SNVs that are different between tumor and normal sample.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://varscan.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">VarScan</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A platform-independent, technology-independent software tool for identifying SNVs, indels, and CNVs in massively parallel sequencing of individual and pooled samples.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">3. Indel Detection</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.sanger.ac.uk/resources/software/dindel/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Dindel</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A program for calling small indels from short-read sequence data from Illumina platform.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="https://trac.nbic.nl/pindel/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Pindel</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for identifying indels and structural variants at single-based resolution from next-generation sequence data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.seqan.de/projects/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SplazerS</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for detecting genomic indel variants with exact breakpoints in single- and paired-end sequencing.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">4. Structural Variation Detection</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://breakdancer.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">BreakDancer</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for detecting five types of SVs (insertions, deletions, inversions, inter- and intra-chromosomal translocations) from next generation paired-end sequencing reads.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="ftp://ftp.stjude.org/pub/software/CREST/CREST.tgz%20%5Ct%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CREST</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software that uses the soft-clipped reads to directly map the breakpoints of SVs.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://code.google.com/p/gasv/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">GASV</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for identifying and comparing structural variants by computing intersections of breakpoint regions.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://code.google.com/p/hydra-sv/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">HYDRA</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for detecting structural variants in both unique and duplicated genomic regions.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://sv.gersteinlab.org/pemer/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">PEMer</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for detecting SVs from paired-end reads.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.bioconductor.org/packages/2.10/bioc/html/R453Plus1Toolbox.html%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">R453Plus1Toolbox</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An R/Bioconductor package for the analysis of Roche 454 sequencing data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://svmerge.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SVMerge</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for SVs analysis by integrating calls from several existing SV callers.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://svdetect.sourceforge.net/Site/Home.html%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SVDetect</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for identifying structural variations from paired-end/mate pair data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://compbio.cs.sfu.ca/strvar.htm%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">VariationHunter</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An tool for identifying structural variations from paired-end WGS data.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">5. Copy Number Variation Detection</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="https://r-forge.r-project.org/R/?group_id=702%20\t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CBS</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An R package for detecting CNVs using sequencing data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="https://dsgweb.wustl.edu/qunyuan/software/cmds/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CMDS</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A population-based method for recurrent CNVs analysis from multiple samples.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.compbio.group.cam.ac.uk/Resources/CNAseg/CNAseg.rar%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CNAseg</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for Identifying CNVs in cancer from NGS data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.wustl.edu/software/cnvhmm%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">cnvHMM</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for CNVs analysis using Hidden Markov algorithm.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://sv.gersteinlab.org/cnvnator/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CNVnator</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for CNV discovery and genotyping from depth of read mapping.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://bioinfo-out.curie.fr/projects/freec/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">FREEC</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for control-free CNVs detection using deep-sequencing data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://sourceforge.net/projects/rdxplorer/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">RDXplorer</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for CNVs detection in whole human genome sequence data using read depth coverage.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&amp;paper_id=182%20\t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">SegSeq</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A tool for detecting CNVs from short sequence reads.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://varscan.sourceforge.net/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">VarScan</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A platform-independent, technology-independent software tool for identifying SNVs, indels, and CNVs in massively parallel sequencing of individual and pooled samples.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">6. Annotation</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.openbioinformatics.org/annovar/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">ANNOVAR</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An efficient software tool to use update-to-date information to functionally annotate genetic variants detected from diverse genomes.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://sv.gersteinlab.org/breakseq/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">BreakSeq</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A pipeline for annotation, classification and analysis of SVs at single nucleotide resolution.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://gvs.gs.washington.edu/SeattleSeqAnnotation/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Seattle Seq</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: An server that provides annotation of SNVs.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">7. Data Visualization</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.strandls.com/Avadis%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Avadis</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software for visualizing and analyzing RNA-Seq data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://circos.ca/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CIRCOS</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for visualizing genomic events.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.broadinstitute.org/igv/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">IGV</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A high-performance visualization tool for interactive exploration of next-generation sequencing data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.wustl.edu/software/pairoscope%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Pairoscope</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for generating diagrams indicating the relationship of paired end sequencing reads, is most useful for visualizing translocations.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.ucsc.edu/cgi-bin/hgGateway%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">UCSC Genome Browser</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A genome browser that provide precise access to sequence and annotation data for any genomic region of specific interest.</span></div></li></ul><div style="min-height: 14pt; text-align: left;"><span style="min-height: 14pt; font-size: 10.5pt; color: rgb(51, 160, 44); font-family: Helvetica; font-weight: bold;">8. Fusion Gene Detection</span></div><ul style="margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; padding-left: 0pt; list-style-type: disc;"><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.strandls.com/Avadis%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Avadis</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software for visualizing and analyzing RNA-Seq data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://circos.ca/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">CIRCOS</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for visualizing genomic events.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://www.broadinstitute.org/igv/%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">IGV</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A high-performance visualization tool for interactive exploration of next-generation sequencing data.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.wustl.edu/software/pairoscope%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">Pairoscope</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A software package for generating diagrams indicating the relationship of paired end sequencing reads, is most useful for visualizing translocations.</span></div></li><li style="margin-left: 24pt; margin-right: 0pt; padding-left: 0pt; text-indent: 0pt; font-size: 10pt;"><div><a href="http://genome.ucsc.edu/cgi-bin/hgGateway%20/t%20_blank" style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica; text-decoration: underline;">UCSC Genome Browser</a><span style="font-size: 10.5pt; color: rgb(68, 68, 68); font-family: Helvetica;">: A genome browser that provide precise access to sequence and annotation data for any genomic region of specific interest</span></div></li></ul></div><div><br/></div></span>
</div></body></html> 