<html>
<head>
  <title>python 编程案例6（anno_geNomePos_from_gff.py）</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="572"/>
<h1>python 编程案例6（anno_geNomePos_from_gff.py）</h1>

<div>
<span><div>python 编程案例6（anno_geNomePos_from_gff.py）</div><div><br/></div><div>#!/share/ofs1a/EXdev/luozw/anaconda3/bin/python</div><div>import sys,os,re,time,argparse,logging,subprocess</div><div><br/></div><div>file_gff=&quot;/share/ofs1a/EXdev/database/hg_19_gff/GCF_000001405.25_GRCh37.p13_genomic.gff&quot;</div><div>hg19_fa=&quot;/share/ofs1a/EXdev/WES_pipe_v4/db/J_hg19_reference/hg19_reference_with_NC.fasta&quot;</div><div>samtools=&quot;/share/ofs1a/EXdev/WES_pipe_v4/bin/samtools&quot;</div><div>NC2chr={&quot;NC_000001&quot;:&quot;chr1&quot;,&quot;NC_000002&quot;:&quot;chr2&quot;,&quot;NC_000003&quot;:&quot;chr3&quot;,&quot;NC_000004&quot;:&quot;chr4&quot;,&quot;NC_000005&quot;:&quot;chr5&quot;,\</div><div>&quot;NC_000006&quot;:&quot;chr6&quot;,&quot;NC_000007&quot;:&quot;chr7&quot;,&quot;NC_000008&quot;:&quot;chr8&quot;,&quot;NC_000009&quot;:&quot;chr9&quot;,&quot;NC_000010&quot;:&quot;chr10&quot;,&quot;NC_000011&quot;:&quot;chr11&quot;,\</div><div>&quot;NC_000012&quot;:&quot;chr12&quot;,&quot;NC_000013&quot;:&quot;chr13&quot;,&quot;NC_000014&quot;:&quot;chr14&quot;,&quot;NC_000015&quot;:&quot;chr15&quot;,&quot;NC_000016&quot;:&quot;chr16&quot;,&quot;NC_000017&quot;:&quot;chr17&quot;,\</div><div>&quot;NC_000018&quot;:&quot;chr18&quot;,&quot;NC_000019&quot;:&quot;chr19&quot;,&quot;NC_000020&quot;:&quot;chr20&quot;,&quot;NC_000021&quot;:&quot;chr21&quot;,&quot;NC_000022&quot;:&quot;chr22&quot;,&quot;NC_000023&quot;:&quot;chrX&quot;,&quot;NC_000024&quot;:&quot;chrY&quot;}</div><div><br/></div><div>HELP=&quot;&quot;&quot;</div><div>    This program is used to get the genome position of some variant from the gff file !</div><div>&quot;&quot;&quot;</div><div><br/></div><div>class _newdict(dict):</div><div>    def __missing__(self, key):</div><div>        value=self[key]=type(self)()    #type(self)() # retain local pointer to value</div><div>        return (value)            # faster to return than dict lookup</div><div><br/></div><div>def _argparses():</div><div>    parser = argparse.ArgumentParser(description='To anno genome position from gff !')</div><div>    parser.add_argument('-g',required=True,dest='gene_pos',help=&quot;gene_position:  GLUD1P7:c.1 (SpecialNotice:Focus only on variations in coding regions !)&quot;)</div><div>    parser.add_argument('-c',required=False,dest='chr_pos',help=&quot;chr_position:  chr1:1000000&quot;)</div><div>    parser.add_argument('-v','--version',required=False,action='store_true',help=&quot;virsion information&quot;)</div><div>    if len(sys.argv) == 1:</div><div>        parser.print_help()</div><div>        sys.exit(0)</div><div>    elif ('-v' in sys.argv) or ('--version' in sys.argv):</div><div>        print(HELP)</div><div>        sys.exit(0)</div><div>    else:</div><div>        return parser.parse_args()</div><div><br/></div><div>def _get_gene_info(gene_name):</div><div>    gene_info=_newdict()</div><div>    rna_name=_newdict()</div><div>    for line in open(file_gff,&quot;r&quot;):</div><div>        if line[0] == &quot;#&quot;:continue</div><div>        line=line.strip(&quot;\n&quot;)</div><div>        line_arr=line.split(&quot;\t&quot;)</div><div>        line_arr[0]=line_arr[0].split(&quot;.&quot;)[0]</div><div>        if (&quot;exon&quot; in line_arr[2]) and (&quot;Parent=&quot; in line_arr[8]) and (&quot;transcript_id=&quot; in line_arr[8]):</div><div>            parent_name=line_arr[8].split(&quot;Parent=&quot;)[1].split(&quot;;&quot;)[0]</div><div>            nm_name=line_arr[8].split(&quot;transcript_id=&quot;)[1].split(&quot;;&quot;)[0]</div><div>            rna_name[parent_name]=nm_name</div><div>        if (not &quot;CDS&quot; in line_arr[2]) or (not &quot;gene=&quot; in line_arr[8]) or (not &quot;protein_id=&quot; in line_arr[8]):continue</div><div>        tmp_gene_name=line_arr[8].split(&quot;gene=&quot;)[1].split(&quot;;&quot;)[0]</div><div>        tmp_np=line_arr[8].split(&quot;protein_id=&quot;)[1].split(&quot;;&quot;)[0]</div><div>        parent_name=line_arr[8].split(&quot;Parent=&quot;)[1].split(&quot;;&quot;)[0]</div><div>        tmp_nm=f&quot;{rna_name[parent_name]}({tmp_np})&quot;</div><div>        if tmp_gene_name == gene_name:</div><div>            if len(gene_info[tmp_nm])==0:</div><div>                gene_info[tmp_nm][&quot;info&quot;]=[]</div><div>                gene_info[tmp_nm][&quot;info&quot;].append([line_arr[0],line_arr[3],line_arr[4]])</div><div>                gene_info[tmp_nm][&quot;count&quot;]=0</div><div>                gene_info[tmp_nm][&quot;strand&quot;]=line_arr[6]</div><div>                gene_info[tmp_nm][&quot;length&quot;]=0</div><div>            else:</div><div>                gene_info[tmp_nm][&quot;info&quot;].append([line_arr[0],line_arr[3],line_arr[4]])</div><div>                gene_info[tmp_nm][&quot;count&quot;]+=1</div><div>                gene_info[tmp_nm][&quot;length&quot;]+=int(line_arr[4])-int(line_arr[3])+1</div><div>    return gene_info</div><div><br/></div><div>def _get_fasta(position):</div><div>    [tmp_chr,tmp_start,*rubish]=re.split(r&quot;:|-&quot;,position)</div><div>    if &quot;NC&quot; in tmp_chr:</div><div>        tmp_chr=NC2chr[tmp_chr]</div><div>    tmp_fa=subprocess.getoutput(f&quot;{samtools} faidx {hg19_fa} {tmp_chr}:{tmp_start}-{tmp_start}|tail -1&quot;)</div><div>    tmp_fa=tmp_fa.strip(&quot;\n&quot;)</div><div>    return tmp_fa</div><div>    </div><div>def _get_cds_info(gene_info,gene_input,gene,origin_cds_pos,origin_intro_pos,pos_index):</div><div>    for nm in sorted(gene_info.keys()):</div><div>        cds_pos=origin_cds_pos</div><div>        nm_count=gene_info[nm][&quot;count&quot;]</div><div>        nm_strand=gene_info[nm][&quot;strand&quot;]</div><div>        cds_length=gene_info[nm][&quot;length&quot;]</div><div>        cds_info=gene_info[nm][&quot;info&quot;]</div><div>        [cds_start,cds_end,cds_num]=[0,0,0]</div><div>        if &quot;-&quot; in nm_strand:</div><div>            for [tmp_chr,tmp_start,tmp_end] in cds_info:</div><div>                cds_num+=1</div><div>                cds_start=cds_end+1</div><div>                cds_end+=int(tmp_end)-int(tmp_start)+1</div><div>                if cds_pos &gt;=cds_start and cds_pos&lt;=cds_end:</div><div>                    real_pos=int(tmp_end)-(cds_pos-cds_start)</div><div>                    if (origin_intro_pos &gt;0) and (&quot;down&quot; in pos_index):</div><div>                        if real_pos==int(tmp_start):</div><div>                            real_pos=real_pos-origin_intro_pos</div><div>                        else:</div><div>                            ###位置不在变体边缘，不是同一个变体</div><div>                            break</div><div>                    elif (origin_intro_pos &gt;0) and (&quot;up&quot; in pos_index):</div><div>                        if real_pos==int(tmp_end):</div><div>                            real_pos=real_pos+origin_intro_pos</div><div>                        else:</div><div>                            ###位置不在变体边缘，不是同一个变体</div><div>                            break</div><div>                    base=_get_fasta(f&quot;{tmp_chr}:{real_pos}&quot;)</div><div>                    print (f&quot;{tmp_chr}:{real_pos}({base})\tcds{cds_num}:{tmp_chr}:{tmp_start}-{tmp_end}\t{nm};tot_cds={nm_count};strand={nm_strand}\t{gene_input}&quot;)</div><div>                    break</div><div>                else:</div><div>                    continue ####只要是在CDS以内,一定在上面的情况里面才对</div><div>        elif &quot;+&quot; in nm_strand:</div><div>            for [tmp_chr,tmp_start,tmp_end] in cds_info:</div><div>                cds_num+=1</div><div>                cds_start=cds_end+1</div><div>                cds_end+=int(tmp_end)-int(tmp_start)+1</div><div>                if cds_pos &gt;=cds_start and cds_pos&lt;=cds_end:</div><div>                    real_pos=cds_pos-cds_start+int(tmp_start)</div><div>                    if (origin_intro_pos &gt;0) and (&quot;down&quot; in pos_index):</div><div>                        if real_pos==int(tmp_end):</div><div>                            real_pos=real_pos+origin_intro_pos</div><div>                        else:</div><div>                            ###位置不在变体边缘，不是同一个变体</div><div>                            break</div><div>                    elif (origin_intro_pos &gt;0) and (&quot;up&quot; in pos_index):</div><div>                        if real_pos==int(tmp_start):</div><div>                            real_pos=real_pos-origin_intro_pos</div><div>                        else:</div><div>                            ###位置不在变体边缘，不是同一个变体</div><div>                            break</div><div>                    base=_get_fasta(f&quot;{tmp_chr}:{real_pos}&quot;)</div><div>                    print (f&quot;{tmp_chr}:{real_pos}({base})\tcds{cds_num}:{tmp_chr}:{tmp_start}-{tmp_end}\t{nm};tot_cds={nm_count};strand={nm_strand}\t{gene_input}&quot;)</div><div>                    break</div><div>                else:</div><div>                    continue ####只要是在CDS以内,一定在上面的情况里面才对</div><div>        else:</div><div>            print (f&quot;error:gene_strand is error:{nm_strand}\n&quot;)</div><div>            sys.exit(0)</div><div><br/></div><div>def _gene_convert(gene_pos):</div><div>    [gene,other_info]=re.split(r&quot;:c.&quot;,gene_pos)</div><div>    gene_info=_get_gene_info(gene)</div><div>    if other_info[0]==r&quot;-&quot;:</div><div>        ###以5'utr定位</div><div>        print (&quot;error:utr in 5'\n&quot;)</div><div>        sys.exit(0)</div><div>    elif other_info[0]==r&quot;*&quot;:</div><div>        ###以3'utr定位</div><div>        print (&quot;error:utr in 3'\n&quot;)</div><div>        sys.exit(0)</div><div>    else:</div><div>        ###以cds定位</div><div>        pos_index=&quot;down&quot;</div><div>        if '+' in other_info:</div><div>            ###3‘内含子</div><div>            [cds_pos,intro_pos]=re.split(r&quot;\+&quot;,other_info)</div><div>            pos_index=&quot;down&quot;</div><div>        elif '-' in other_info:</div><div>            ###5'内含子</div><div>            [cds_pos,intro_pos]=re.split(r&quot;\-&quot;,other_info)</div><div>            pos_index=&quot;up&quot;</div><div>        else:</div><div>            ###在cds上</div><div>            [cds_pos,intro_pos]=[other_info,0]</div><div>            pos_index=&quot;down&quot;</div><div>#        print (f&quot;{cds_pos}\t{intro_pos}\t{pos_index}&quot;)</div><div>#        sys.exit(0)</div><div>        _get_cds_info(gene_info,gene_pos,gene,int(cds_pos),int(intro_pos),pos_index)</div><div><br/></div><div>def _chr_convert(chr_pos):</div><div>    print (&quot;here is chr_convert!&quot;)</div><div><br/></div><div>def _main():</div><div>    logging.basicConfig(format='%(levelname)s %(asctime)s,line %(lineno)s:\t%(message)s',level=logging.DEBUG,datefmt='%Y-%m-%d %H:%M:%S')</div><div>    argvs = _argparses()</div><div>#    print (argvs)</div><div>#    logging.info(&quot;work start&quot;)</div><div>    if argvs.gene_pos != None:</div><div>        _gene_convert(argvs.gene_pos)</div><div>    elif argvs.chr_pos != None:</div><div>        _chr_convert(argvs.chr_pos)</div><div>    else:</div><div>        print (&quot;error:input error !!!\n&quot;)</div><div><br/></div><div>if __name__ == '__main__':</div><div>    _main()</div><div><br/></div><div><br/></div></span>
</div></body></html> 