<html>
<head>
  <title>tumor 【WGS、TAS】 CNV 分析工具之：CNVkit</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1232"/>
<h1>tumor 【WGS、TAS】 CNV 分析工具之：CNVkit</h1>

<div>
<span><div><h1 style="word-wrap: break-word; margin: 0px 0px 0px 10px; padding: 0px; border: 0px; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><span style="word-wrap: break-word; border: 0px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><span style="font-size: 16pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; line-height: 48px;">tumor 【WGS、TAS】 CNV 分析工具之：CNVkit</span></span></h1><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">1）CNVkit 安装</span></font></div><div><span style="font-size: 10pt;">    git-hub地址：https://github.com/etal/cnvkit （git clone</span> <a href="https://github.com/etal/cnvkit" style="font-size: 10pt;">https://github.com/etal/cnvkit</a><span style="font-size: 10pt;">）</span></div><div><span style="font-size: 10pt;">    应用示例：</span><a href="http://www.dxy.cn/bbs/topic/38832790?sf=2&amp;dn=8" style="font-size: 10pt;">http://www.dxy.cn/bbs/topic/38832790?sf=2&amp;dn=8</a></div><div><span style="font-size: 10pt; color: rgb(255, 0, 0);">    说明文档：</span><a href="https://cnvkit.readthedocs.io/en/stable/" style="font-size: 10pt; color: rgb(255, 0, 0);">https://cnvkit.readthedocs.io/en/stable/</a></div><div><span style="font-size: 10pt;">    安装：conda create -n cnvkit cnvkit    或者  conda install cnvkit</span></div><div><span style="font-size: 10pt;">    依赖Python包：Biopython,</span><span style="font-size: 10pt; font-weight: bold;">Reportlab</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">matplotlib</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">NumPy</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">SciPy</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">Pandas</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">pyfaidx</span><span style="font-size: 10pt;">,</span><span style="font-size: 10pt; font-weight: bold;">pysam</span></div><div><span style="font-size: 10pt;">    依赖R包：library(&quot;BiocManager&quot;)，install.packages(&quot;DNAcopy&quot;)</span></div><div><span style="font-size: 10pt;">    文献参考：</span>CNVkit: Genome-Wide Copy Number Detection and Visualization from Targeted DNA Sequencing</div><div>    安装完成：/mnt/X500/farmers/luozw/bin/anaconda/envs/bioinfoLzw/bin/cnvkit.py</div><div><span style="font-size: 10pt;">    分析目标：</span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">CNVkit is primarily designed for use on</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="box-sizing: border-box; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(255, 0, 0); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">hybrid capture</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">sequencing data, where off-target reads are present and can be used improve copy number estimates. However, CNVkit can also be used on</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="box-sizing: border-box; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(255, 0, 0); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">whole-genome sequencing</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(255, 0, 0); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">(WGS) and</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(255, 0, 0); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="box-sizing: border-box; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(255, 0, 0); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">targeted amplicon sequencing</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252); font-size: 10pt; color: rgb(64, 64, 64); font-family: Lato, proxima-nova, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">(TAS) datasets by using alternative command-line options.</span></div><div><br/></div><div><br/></div><div><span style="color: rgb(255, 0, 0);">    其他：</span><a href="https://shahlab.ca/projects/TitanCNA/" style="color: rgb(255, 0, 0);">https://shahlab.ca/projects/TitanCNA/</a><span style="color: rgb(255, 0, 0);">  这个工程下面还有一些工具，有空可以瞅瞅。</span></div><div><span style="font-size: 12pt;">2）使用说明</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">    Python cnvkit.py batch  [tumor_bam_file]  </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">【args】</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">      </span> <span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;"> </span><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">-m,--method</span> <span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">  </span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">     hybrid | amplicon | wgs</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                                          hybrid:    hybridization capture</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                                          amplicon:    targeted amplicon sequencing</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                                          </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">wgs:    whole genome sequencing</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        -y, --male-reference, --haploid-x-reference   </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">要不要区分男女？？？</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        -c, --count-reads        Get read depths by counting read midpoints within each bin. (An alternative algorithm).</span></div><div><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">        --drop-low-coverage    Drop very-low-coverage bins before segmentation to avoid false-positive deletions in poor-quality tumor samples.（去掉覆盖差的区域）</span></div><div><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">         -p , --processes [PROCESSES]    指定子线程数用以平行分析两个bam，否则串行。</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                e.g.  4</span></div><div><span style="font-size: 11pt;"><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">        --rscript-path PATH    Path to the Rscript excecutable to use for running R code。[Default: Rscript] 指定Rscript 路径</span></span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                e.g.  </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">/mnt/X500/farmers/luozw/bin/anaconda/envs/bioinfoLzw/bin/Rscript</span></div><div><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">        -n , --normal [FILES [FILES ...]]    Normal samples bam</span></div><div><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">        -f --fasta FILENAME        Reference genome, FASTA format (e.g. UCSC hg19.fa)</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                 e.g.   </span><span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: pre-wrap; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-family: &quot;Microsoft YaHei&quot;, &quot;Segoe UI&quot;, system-ui, Roboto, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif, Tahoma, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">/mnt/X500/farmers/linhx/bin/ref/hs37d5.fa</span></div><div><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">         -t FILENAME, --targets FILENAME    Target intervals (.bed or .list) </span></div><div>                              ( The BED file should be the baited genomic regions for your target capture kit, as provided by your vendor)</div><div>                              ( The batch <span style="color: rgb(255, 0, 0);">--method wgs option uses the given reference genome’s sequencing-accessible regions</span> (“access” BED) as the “targets” )</div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">     </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">    --annotate FILENAME        Use gene models from this file to assign names to the target regions. </span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                                                        Format: UCSC refFlat.txt or ensFlat.txt file (preferred), or BED, interval list, GFF, or similar</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                 .e.g.    </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">/mnt/X500/farmers/luozw/factory/20190821_cnvkit_test/Homo_sapiens.GRCh37.75.gtfTransTo.gff3</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">         --short-names        Reduce multi-accession bait labels to be short and consistent.</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">         </span><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">--target-avg-size    Average size of split target bins</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                             </span>           (  at least 1000 bases for 30x coverage, or proportionally more for lower-coverage sequencing.)</div><div><span style="color: rgb(255, 0, 0);">                e.g. 6X-WGS，最少5Kb，（和经验值差不多，0,6X-50K，6X-5K）</span></div><div><span style="font-size: 11pt;"><span style="font-size: 11pt; color: rgb(227, 0, 0); font-family: &quot;Times New Roman&quot;;">         -g FILENAME, --access FILENAME  Regions of accessible sequence on chromosomes (.bed), as output by the 'access' command.</span></span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                e.g.  </span><span style="font-family: &quot;Times New Roman&quot;;"><span style="font-size: 11pt;">/mnt/X500/farmers/luozw/factory/20190821_cnvkit_test/hs37d5.fa_target.bed</span> <font color="#FF0000" style="font-size: 14pt;"><b>(</b></font></span><font color="#FF0000" style="font-size: 14pt;"><b><span style="font-family: &quot;Times New Roman&quot;;">cnvkit.py access hs37d5.fa -o hs37d5.</span><span style="font-family: &quot;Times New Roman&quot;;">fa_target</span><span style="font-family: &quot;Times New Roman&quot;;">.bed</span><span style="font-family: &quot;Times New Roman&quot;;">)</span></b></font></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">      </span> <span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">  --antitarget-avg-size        ANTITARGET_AVG_SIZE  Average size of antitarget bins</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                e.g. WGS 无</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">antitarget区域</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">         --antitarget-min-size        ANTITARGET_MIN_SIZE    Minimum size of antitarget bins</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">                </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">e.g. WGS 无</span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">antitarget区域</span></div><div><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">         -d DIRECTORY, --output-dir DIRECTORY        Output directory</span></div><div><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">         --scatter        Create a whole-genome copy ratio profile as a PDF scatter plot</span></div><div><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">         --diagram      Create an ideogram of copy ratios on chromosomes as a PDF.</span></div><div><br/></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">        </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">To reuse an existing reference: (<span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;; color: rgb(255, 0, 0);">指定先前样本outdir生成的</span></span><span style="font-size: 11pt;"><span style="font-size: 11pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">reference.cnn 为对照样本进行分析，节约时间。</span></span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">)</span></div><div><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">         -r REFERENCE, --reference REFERENCE   </span><span style="font-size: 11pt; font-family: &quot;Times New Roman&quot;;">Copy number reference file (.cnn).</span></div><div>        如： <span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-family: &quot;PingFang SC&quot;, &quot;Source Han Sans SC&quot;, &quot;Noto Sans CJK SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei UI&quot;, &quot;Microsoft YaHei&quot;, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">/home/anaconda2/envs/cnvkit/bin/cnvkit.py batch 704_bwa.sam.bam -r my_flat_reference.cnn -d 704 -p 10</span></div><div><span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></span></div><div><span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></span></div><div><span style="font-size: 14px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></span></div><div><br/></div><div><br/></div></div><div><br/></div></span>
</div></body></html> 