<html>
<head>
  <title>tumor 【超低深度WGS】定量分析与CNA分析之ichorCNA：</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1277"/>
<h1>tumor 【超低深度WGS】定量分析与CNA分析之ichorCNA：</h1>

<div>
<span><div><span style="font-size: 16pt; color: rgb(255, 0, 0);">tumor 【</span><span style="font-size: 16pt; color: rgb(255, 0, 0);">超</span><span style="font-size: 16pt; color: rgb(255, 0, 0); font-weight: bold;">低深度WGS】</span><span style="font-size: 16pt; color: rgb(255, 0, 0);">定量分析与CNA分析之</span><span style="font-size: 16pt; color: rgb(255, 0, 0); font-weight: bold;">ichorCNA：</span></div><div><font face="Times New Roman" style="font-size: 10pt;"><br/></font></div><div><font style="font-size: 14pt;"><span style="font-size: 14pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">Description</span></font></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">    ichorCNA is a tool for estimating the fraction of tumor in cell-free DNA from ultra-low-pass whole genome sequencing. </span><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;"> ichorCNA can be used to</span> <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">inform the presence or absence of tumor-derived DNA</span> <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">and to guide the decision to</span> <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">perform whole exome or deeper whole genome sequencing</span><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">. Furthermore, the quantitative estimate of tumor fraction can we used to calibrate the desired depth of sequencing to reach statistical power for identifying mutations in cell-free DNA. Finally, ichorCNA can be use to</span> <span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">detect large-scale copy number alterations</span> <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">from large cohorts by taking advantage of the cost-effective approach of ultra-low-pass sequencing.</span></div><div><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 252, 252);"><br/></span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">1）  软件官网：</span><a href="https://labworm.com/tool/ichorcna" style="font-size: 12pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">https://labworm.com/tool/ichorcna</a></div><div>        文献： <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">Adalsteinsson, Ha, Freeman, et al. Scalable whole-exome sequencing of cell-free DNA reveals high concordance with metastatic tumors.</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        git-hub地址：</span><a href="https://github.com/broadinstitute/ichorCNA/wiki?ref=labworm" style="font-size: 12pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;;">https://github.com/broadinstitute/ichorCNA/wiki?ref=labworm</a></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        参数详细说明：</span><span style="font-size: 12pt;"><a href="https://github.com/broadinstitute/ichorCNA/wiki/Create-Panel-of-Normals" style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">https://github.com/broadinstitute/ichorCNA/wiki/Create-Panel-of-Normals</a></span></div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">2）安装：</span></font></div><div>        2.1)Using R devtools</div><div>            install.packages(&quot;devtools&quot;)</div><div>            library(devtools)</div><div>            install_github(&quot;broadinstitute/ichorCNA&quot;)</div><div><span style="font-weight: bold;">            option 2 </span> <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">Manual installation</span></div><div>        2.2) 安装hmmcopy_utils</div><div>            github地址：<a href="https://github.com/shahcompbio/hmmcopy_utils">https://github.com/shahcompbio/hmmcopy_utils</a></div><div>            本地安装地址：/GP/work/luozw/database/git-hub/hmmcopy_utils/bin/</div><div>                    gcCounter    --&gt;        gcCounter -w 100000 -c 1,3,5,X hg18.fasta &gt; hg18.gc.wig</div><div>                    mapCounter    --&gt;     mapCounter -w 100000 -c 1,3,5,X hg18.bw &gt; hg18.map.wig</div><div>                    readCounter    --&gt;    readCounter -w 100 -c 1,3,5,X aligned_reads.bam &gt; readcounts.wig</div><div><font style="color: rgb(227, 0, 0); font-size: 12pt;"><b>        2.3) 本地运行地址：/mnt/X500/farmers/luozw/factory/20190821_15WGS_sv_cnv/cnv_result/ichorCNA_50K/cal_ichorCNA_by_pon_mean.sh</b></font><br/></div><div><br/></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">3）使用方法：</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">    step-1）用readCounter生成wig 文件</span></div><div><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">         </span><span style="font-size: 12pt;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">readCounter -w 50000 -q 30 -c &quot;1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y&quot; $normal_bam &gt; $out_dir/normal.wig</span></span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">    </span><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">step-2）分析作图</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">   <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold; color: rgb(255, 0, 0);"> </span><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold; color: rgb(255, 0, 0);">     </span></span><span style="font-size: 12pt;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">source activate bioinfoLzw</span></span></div><div><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">    </span><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">     Rscript /mnt/X500/farmers/luozw/bin/ichorCNA/scripts/runIchorCNA.R \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --id ${normal_id}_${tumor_id}_pon \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --WIG $out_dir/tumor.wig \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --NORMWIG $out_dir/normal.wig \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --ploidy &quot;c(2,3)&quot; \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --normal &quot;c(0.5,0.6,0.7,0.8,0.9)&quot; \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --maxCN 5  \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --gcWig /mnt/X500/farmers/luozw/factory/20190821_15WGS_sv_cnv/cnv_result/ichorCNA_50K/gc_hg19_50kb.wig \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --mapWig /mnt/X500/farmers/luozw/factory/20190821_15WGS_sv_cnv/cnv_result/ichorCNA_50K/map_hg19_50kb.wig  \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --centromere /mnt/X500/farmers/linhx/bin/miniconda3/envs/py37_withR/share/r-ichorcna-0.1.0.20180710-0/extdata/GRCh37.p13_centromere_UCSC-gapTable.txt \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --rmCentromereFlankLength 1e5 \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --includeHOMD False \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --chrs &quot;c(1:22, \&quot;X\&quot;)&quot; \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --chrTrain &quot;c(1:22)&quot; \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --estimateNormal True \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --estimatePloidy True \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --estimateScPrevalence True \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --scStates &quot;c(1,3)&quot; \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --txnE 0.9999999 \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --txnStrength 1e7 \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --outDir $out_dir \</span></div><div><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">        --normalPanel /mnt/X500/farmers/luozw/factory/20190821_15WGS_sv_cnv/cnv_result/ichorCNA_50K/PON/pon_out_file_median_median.rds</span></div><div><br/></div><div><span style="font-size: 12pt;"><span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;; font-weight: bold;">    参数详解）</span></span></div><div>    Rscript runIchorCNA.R --help</div><div>    Usage: runIchorCNA.R [options]</div><div>    Options:</div><div>    #基本参数</div><div>        --id=ID        Patient ID. Default: [test]</div><div>        --WIG=WIG        Path to tumor WIG file. Required.</div><div>        --NORMWIG=NORMWIG        Path to normal WIG file. Default: [NULL]</div><div>        --gcWig=GCWIG        Path to GC-content WIG file; Required</div><div>        --mapWig=MAPWIG        Path to mappability score WIG file. Default: [NULL]</div><div>        --centromere=CENTROMERE        File containing Centromere locations; if not provided then will use hg19 version from ichorCNA package. Default: [NULL]</div><div>        --outDir=OUTDIR        Output Directory. Default: [./]</div><div>        --chrTrain=CHRTRAIN        Specify chromosomes to estimate params. Default: [c(1:22)]</div><div>        --chrs=CHRS        Specify chromosomes to analyze. Default: [c(1:22,&quot;X&quot;)]</div><div>        --ploidy=PLOIDY        Initial tumour ploidy; can be more than one value if additional ploidy initializations are desired. Default: [2]</div><div>        --maxCN=MAXCN        Total clonal CN states. Default: [7]</div><div>        --normal=NORMAL        c(0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,0.95,0.99)：Initialize the non-tumor (1 minus tumor fraction) to expected values</div><div>        --exons.bed=EXONS.BED        Path to bed file containing exon regions. Default: [NULL]</div><div>        --estimateNormal=ESTIMATENORMAL    Estimate normal. Default: [TRUE]</div><div>        --estimateScPrevalence=ESTIMATESCPREVALENCE    Estimate subclonal prevalence. Default: [TRUE]</div><div>        --estimatePloidy=ESTIMATEPLOIDY        Estimate tumour ploidy. Default: [TRUE]</div><div>        --chrNormalize=CHRNORMALIZE        Specify chromosomes to normalize GC/mappability biases. Default: [c(1:22)]</div><div>        --plotYLim=PLOTYLIM        ylim to use for chromosome plots. Default: [c(-2,2)]</div><div><br/></div><div>    #可选参数</div><div><span style="color: rgb(255, 0, 0); font-weight: bold;">        --normalPanel=NORMALPANEL         Median corrected depth from panel of normals. Default: [NULL]</span></div><div>        --scStates=SCSTATES        Subclonal states to consider. Default: [NULL] c(1,3)</div><div><br/></div><div>    #非必要参数</div><div>        --rmCentromereFlankLength=RMCENTROMEREFLANKLENGTH    Length of region flanking centromere to remove. Default: [1e+05]</div><div>        --coverage=COVERAGE        PICARD sequencing coverage. Default: [NULL]</div><div>        --lambda=LAMBDA        Initial Student's t precision; must contain 4 values (e.g. c(1500,1500,1500,1500)); if not provided then will automatically use based on variance of data. Default: [NULL]</div><div>        --lambdaScaleHyperParam=LAMBDASCALEHYPERPARAM        Hyperparameter (scale) for Gamma prior on Student's-t precision. Default: [3]</div><div>        --maxFracCNASubclone=MAXFRACCNASUBCLONE        Exclude solutions with fraction of subclonal events greater than this value. Default: [0.7]</div><div>        --maxFracGenomeSubclone=MAXFRACGENOMESUBCLONE        Exclude solutions with subclonal genome fraction greater than this value. Default: [0.5]</div><div>        --minSegmentBins=MINSEGMENTBINS        Minimum number of bins for largest segment threshold required to estimate tumor fraction; if below this threshold, then will be assigned zero tumor fraction.</div><div>        --altFracThreshold=ALTFRACTHRESHOLD        Minimum proportion of bins altered required to estimate tumor fraction; if below this threshold, then will be assigned zero tumor fraction. Default: [0.05]</div><div>        --normalizeMaleX=NORMALIZEMALEX        If male, then normalize chrX by median. Default: [TRUE]</div><div>        --fracReadsInChrYForMale=FRACREADSINCHRYFORMALE        Threshold for fraction of reads in chrY to assign as male. Default: [0.001]</div><div>        --includeHOMD=INCLUDEHOMD    If FALSE, then exclude HOMD state. Useful when using large bins (e.g. 1Mb). Default: [FALSE]</div><div>        --txnE=TXNE    Self-transition probability. Increase to decrease number of segments. Default: [0.9999999]</div><div>        --txnStrength=TXNSTRENGTH        Transition pseudo-counts. Exponent should be the same as the number of decimal places of --txnE. Default: [1e+07]</div><div>        --plotFileType=PLOTFILETYPE    File format for output plots. Default: [pdf]</div><div>        --libdir=LIBDIR        Script library path. Usually exclude this argument unless custom modifications have been made to the ichorCNA R package code and the user would like to source those R files. Default: [NULL]</div><div><br/></div><div><span style="font-size: 12pt; font-weight: bold;">4）结果解读</span></div><div>       1） Benchmarking of ichorCNA using metastatic breast/prostate patient <span style="color: rgb(255, 0, 0); font-weight: bold;">cfDNA</span> and healthy donor cfDNA reveals a lower limit of sensitivity for detecting the presence of tumor to be 0.03 TFx (3%). That is, an <span style="color: rgb(255, 0, 0); font-weight: bold;">ichorCNA-estimate of &gt; 0.03 TFx</span>（ Tumor fraction） will reliable indicate the presence of tumor for a cfDNA sample sequenced to ~0.1x whole genome coverage.</div><div>        2） When selecting cfDNA samples for <span style="color: rgb(255, 0, 0); font-weight: bold;">standard depths of</span> <span style="color: rgb(255, 0, 0); font-weight: bold;">whole exome sequencing</span> (e.g. <span style="color: rgb(255, 0, 0); font-weight: bold;">mean target coverage ~150x</span>), we recommend samples with <span style="color: rgb(255, 0, 0); font-weight: bold;">ichorCNA estimates of &gt; 0.1 TFx (10%)</span>. From the benchmarking, ichorCNA correctly estimated samples with &gt; 0.1 TFx for 91% (606/669 patient-healthy donor mixtures) and correctly estimated samples with &lt; 0.1 TFx for 96% (613/641 mixtures).ichorCNA has been tuned to be conservative and may sometimes underestimate TFx. This was a design decision that this leads to some samples with <span style="color: rgb(255, 0, 0); font-weight: bold;">0.05-0.09</span> estimated TFx that <span style="color: rgb(255, 0, 0);">may still be suitable for standard depth whole exome sequencing.</span></div><div>        3） <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"><span style="font-size: 16px; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; color: rgb(36, 41, 46);">Samples that have an</span> <span style="font-size: 16px; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; color: rgb(255, 0, 0); font-weight: bold;">estimated tumor fraction ~0.03-0.1 may require manual curation to confirm this estimate.</span></span></div><div>        4） To restrict your results to only “high quality” data we suggest only using samples with a GC-Map correction <span style="color: rgb(255, 0, 0); font-weight: bold;">MAD less than 0.15</span> and also ensuring <span style="color: rgb(255, 0, 0);">the mean coverage (without PCR duplicates) is at least 0.1x</span> (not calculated by ichorCNA).</div><div><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><br/></span></div><div><span style="font-size: 12pt; font-weight: bold;">    </span> <span style="font-size: 13.6px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgba(27, 31, 35, 0.05); color: rgb(36, 41, 46); font-family: SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-variant-caps: normal; font-variant-ligatures: normal;">&lt;sampleID&gt;.params.txt    </span><span style="font-size: 16px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;;">Final converged parameters for optimal solution. Also contains table of converged parameters for all solutions.</span></div><div><span style="font-size: 16px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;;">            </span> <span style="box-sizing: border-box; font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">GC-Map correction MAD：</span> <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">A value greater than 0.3 usually indicates a high variance in the data and it is likely too noisy。</span></div><div><span style="font-size: 16px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;;">    </span> <span style="font-size: 13.6px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgba(27, 31, 35, 0.05); color: rgb(36, 41, 46); font-family: SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-variant-caps: normal; font-variant-ligatures: normal;">&lt;sampleID&gt;/&lt;sampleID&gt;_genomeWide.pdf    </span><span style="background-color: rgb(255, 255, 255); font-size: 16px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;;">Genome wide plot of data annotated for estimated copy number, tumor fraction, and ploidy for the optimal solution.</span></div><div>                1 copy = dark green</div><div>                2 copies = blue</div><div>                3 copies = brown</div><div>                4+ copies = red</div><div><br/></div><div><span style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">5）其它</span></span></div><div>    4.1）wig、bigWig和bedgraph文件详解：<a href="http://www.bio-info-trainee.com/1815.html">http://www.bio-info-trainee.com/1815.html</a></div><div>    4.2） <span style="color: rgb(255, 0, 0); font-weight: bold;">平均绝对误差： Median absolute deviation (</span><span style="color: rgb(255, 0, 0); font-weight: bold;">MAE）， Median Absolute Difference（MAD）</span></div><div>             对同一物理量进行多次测量时，各次测量值及其绝对误差不会相同，我们将各次测量的绝对误差取绝对值后再求平均值，并称其为平均绝对误差，即:△=(│△1│+│△2│+……+│△n│)/n (△为平均绝对误差;△1、△2、……△n为各次测量的绝对误差)。</div><div><br/></div><div><font style="font-size: 12pt;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-weight: bold;">6）PON对照文件生成</span></font></div><div>      Rscript /mnt/X500/farmers/luozw/bin/ichorCNA/scripts/createPanelOfNormals.R \</div><div>        --filelist pon_list \</div><div>        --gcWig /mnt/X500/farmers/luozw/bin/ichorCNA/inst/extdata/gc_hg19_50kb.wig \</div><div>        --mapWig /mnt/X500/farmers/luozw/bin/ichorCNA/inst/extdata/map_hg19_50kb.wig \</div><div>        --centromere /mnt/X500/farmers/luozw/bin/ichorCNA/inst/extdata/GRCh37.p13_centromere_UCSC-gapTable.txt \</div><div>        --method median \</div><div>        --outfile pon_out_file</div><div>    得到文件：</div><div><span style="color: rgb(255, 0, 0); font-weight: bold;">                ./pon_out_file_median.rds        (</span> <span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">pass the</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="box-sizing: border-box; font-size: 13.6px; background-color: rgba(27, 31, 35, 0.05); border-radius: 3px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">.rds</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">output file in using the</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="box-sizing: border-box; font-size: 13.6px; background-color: rgba(27, 31, 35, 0.05); border-radius: 3px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">--normalPanel</span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;"> </span><span style="font-size: 16px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(255, 0, 0); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; font-variant-caps: normal; font-variant-ligatures: normal; font-weight: bold;">option</span><span style="color: rgb(255, 0, 0); font-weight: bold;">)</span></div><div><span style="color: rgb(255, 0, 0); font-weight: bold;">                ./pon_out_file_median.txt</span></div><div><br/></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">When you run ichorCNA, you can pass the .rds output file in using the --normalPanel option to normalize your sample using the PoN.</span></div></span>
</div></body></html> 