<html>
<head>
  <title>WGS-SV 分析工具之：Meerkat</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="1238"/>
<h1>WGS-SV 分析工具之：Meerkat</h1>

<div>
<span><div><span style="word-wrap: break-word; border: 0px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 16pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; line-height: 48px;">WGS-SV 分析工具之：</span><span style="font-size: 16pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;;">Meerkat</span></div><div><span style="font-size: 14pt; font-family: &quot;Times New Roman&quot;;">1）下载安装：</span></div><div><span style="font-size: 10pt;">        官网地址：</span><a href="http://compbio.med.harvard.edu/Meerkat/" style="font-size: 10pt;">http://compbio.med.harvard.edu/Meerkat/</a></div><div><span style="font-size: 10pt;">        参考文献： Diverse mechanisms of somatic structural variations in human cancer genomes</span></div><div><span style="font-size: 10pt;">        下载地址：</span><a href="http://compbio.med.harvard.edu/Meerkat/meerkat_download_page.php" style="font-size: 10pt;">http://compbio.med.harvard.edu/Meerkat/meerkat_download_page.php</a>  <span style="color: rgb(255, 0, 0); font-weight: bold;">(这里有详细的PDF 说明文档)</span></div><div><span style="font-size: 10pt;">        应用示例：<a href="https://www.cnblogs.com/raisok/p/10918187.html" style="font-size: 10pt;">https://www.cnblogs.com/raisok/p/10918187.html</a></span></div><div><span style="font-size: 10pt;">                        </span><a href="https://www.cnblogs.com/ywliao/p/6438732.html" style="font-size: 10pt;">https://www.cnblogs.com/ywliao/p/6438732.html</a></div><div><span style="font-size: 14pt;">2）用法说明：</span></div><div><span style="font-size: 14pt; font-weight: bold;">         2.1) 数据文件配置与软件配置：</span><span style="font-size: 14pt;">    </span></div><div>              Reference file and other files:</div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                1. Reference genome sequence in fasta format and indexed by PERL module Bio::DB::Fasta.The fasta file of reference must be in a folder alone without any other files, see         Bio::DB::Fasta manual for more detail. Reference sequence in 2bit format is required to setup blat server.</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                2. BWA index of reference file generated by “bwa index” command.</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                3. Fai index of reference file generated by “samtools faidx” command.</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                4. Gene annotation of reference genome which can be downloaded from UCSC. The file needs to be sorted by chromosome and coordinate by command “sort refGene.txt -k 3,3 -k 5,5n &gt; refGene_sorted.txt”.</span></div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">                5. Repeat annotation </span><span style="min-height: 15pt; font-size: 10pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">(RepeatMasker output) of reference genome which can be downloaded</span></div><div><span style="min-height: 15pt;"><span style="font-size: 10pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">from UCSC.</span></span></div><div><font style="font-size: 14pt;"><span style="min-height: 15pt;"><span style="font-size: 14pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">            </span></span><span style="font-size: 14pt; font-weight: bold;">2.2) </span><span style="min-height: 15pt; font-size: 14pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">Input file:</span></font></div><div>                <span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">Paired end reads alignment in BAM format.</span></div><div><font face="Times New Roman" style="font-size: 10pt;"><span style="min-height: 15pt; font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(1, 1, 1);">                    1）</span><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(1, 1, 1);">For human 75bp or 100bp reads, recommend “bwa aln” command to use “-l 40 -k 2” and “bwa sampe” command to use “-N 100” parameters.</span></font></div><div><span style="font-size: 10pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">                    2）BAM file must be sorted and indexed. If the genome is aligned by BWA mem, it has to be processed by Picard or other tools to mark duplicates, see section IV and VI.</span></div><div><span style="font-size: 10pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">    </span><span style="font-size: 10pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">        </span><span style="font-size: 14pt; color: rgb(255, 0, 0); font-family: &quot;Times New Roman&quot;; font-weight: bold;">     </span><span style="min-height: 15pt;"><span style="font-size: 14pt; color: rgb(255, 0, 0); font-family: LiberationSerif; font-weight: bold;">using default parameters is not a good idea.</span></span></div><div><span style="min-height: 15pt;"><span style="font-size: 14pt; color: rgb(255, 0, 0); font-family: LiberationSerif; font-weight: bold;">        </span></span><span style="font-size: 14pt; min-height: 15pt; color: rgb(1, 1, 1); font-family: &quot;Times New Roman&quot;;">  </span><span style="font-size: 14pt; font-weight: bold;">2.3) WORK FLOW</span><span style="font-size: 14pt; min-height: 15pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">:</span></div><div><span style="font-size: 14pt; min-height: 15pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">            </span><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">1). pre_process.pl</span></div><div style="margin-left: 40px;">          <span style="color: rgb(255, 0, 0);">  -b FILE    sorted and indexed bam file, required</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">                        input.bam</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0); font-weight: bold;">            -</span><span style="color: rgb(255, 0, 0); font-weight: bold;">k INT </span><span style="color: rgb(255, 0, 0); font-weight: bold;">   min coverage required for a nucleotide to be ignored (ignore position that covered by too many reads), 0 turn this function off, default 500</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0); font-weight: bold;">                        1500</span></div><div style="margin-left: 40px;">            -r INT    range of insert size to be plotted, insert size larger than such won't be used calculating distribution, default 1000</div><div style="margin-left: 40px;">            -n INT    number of reads per read group to be used calculating insert size distribution, 0 use all reads, default 10000</div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">            -l INT    [0/1], re-map read pairs of softclipped reads and unmapped reads, default 1</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0); font-weight: bold;">                        0  #不进行重新比对</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0); font-weight: bold;">                        </span><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif; font-weight: bold;">It can greatly increase sensitivity if insert size is small and read is long (75bp or </span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif; font-weight: bold;">100bp). For short read length (50bp or shorter), set this option to 0. For BWA mem aligned genomes,</span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif; font-weight: bold;">re-mapping is unnecessary, so turn this function off and also turn it off in meerkat.pl.</span></div><div style="margin-left: 40px;"> <span style="color: rgb(255, 0, 0);">           -q INT    Read trimming parameter. Equivalent to BWA's -q option, default 15</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">                        15</span></div><div style="margin-left: 40px;">            -c INT    bp to be cut off from beginning and end of unmapped and soft clipped reads, such reads were used to find smaller events, default 35</div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">            -s INT    bp to be cut off from beginning and end of unmapped and soft clipped reads, such reads were used in split reads mapping, must be same as -s in meerkat.pl, default 20</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">                        20</span></div><div style="margin-left: 40px;">            -u INT    [0/1], process uu pair (both reads unmapped in a pair) into 4 pairs, default 0</div><div style="margin-left: 40px;">                            0</div><div style="margin-left: 40px;"><span style="color: rgb(88, 152, 255);">            -f INT    number of alternative mappings to print in XA tag for clipped alignments, default 100</span></div><div style="margin-left: 40px;"><span style="color: rgb(88, 152, 255);">                        ###为啥不考虑</span></div><div style="margin-left: 40px;">            -N INT    Clipped reads and split reads must have &lt;= INT Ns, default 5</div><div style="margin-left: 40px;">            -t INT    number of threads used in bwa alignment, default 1</div><div style="margin-left: 40px;">            -R FILE    file name of read group to be ignored, one read group ID per line</div><div style="margin-left: 40px;">            -I STR    /path/to/reference/bwa_index for bwa alignment, <span style="font-weight: bold;">required if -l 1       </span> <span style="color: rgb(255, 0, 0); font-weight: bold;">（不是L）</span></div><div style="margin-left: 40px;">            -A STR    /path/to/reference/fasta.fai for bwa alignment, <span style="font-weight: bold;">required if -l 1</span></div><div style="margin-left: 40px;">            -S STR    /path/to/samtools, path only, not the command, <span style="font-weight: bold;">no need to specify if samtools is in PATH</span></div><div style="margin-left: 40px;">            -W STR    /path/to/bwa, path only, not the command, <span style="font-weight: bold;">no need to specify if bwa is in PATH</span></div><div style="margin-left: 40px;"><span style="color: rgb(255, 0, 0);">            -P STR    specify step to run, [all|is|cl], default all</span></div><div style="margin-left: 40px;">                is: extract unmapped, soft clipped reads, calculate insert size distribution</div><div style="margin-left: 40px;">                cl: map soft clipped read pairs to reference genome</div><div style="margin-left: 40px;">              <span style="color: rgb(255, 0, 0);">  all: run all above steps</span></div><div>                    ####<span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">For example, </span></div><div><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">                    for 50bp reads, &lt;10x TCGA genomes, suggest to use “-s 18 -l 0 -q 0”. </span></div><div><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">                    For 75-</span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">101bp reads, 20-30x and 60-80x TCGA genomes, suggest to use “-s 20 -k 1500 -q 15”. </span></div><div><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: LiberationSerif;">                    For BWA mem </span><span style="font-size: 12pt; color: rgb(255, 0, 0); font-family: LiberationSerif;">aligned genomes, use “-s 20 -k 1500 -q 15 -l 0”.</span></div><div><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">                    For NA18507 run in our original publication, use all </span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">defaults. </span></div><div><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">                    For TCGA WES data, use “-s 20 -k 10000 -q 5”.</span></div><div><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">                </span><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">2）.</span> <span style="font-size: 12pt;"><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">perl ./scripts/meerkat.pl [options]</span></span></div><div style="margin-left: 80px;">  <span style="color: rgb(255, 0, 0);">  -b FILE    sorted and indexed bam file, required</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">                inputbam</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -k INT    [0/1], use black list generated in pre_process.pl, default 1</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">                1</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -d FLT    standard deviation cutoff to call discordant read pairs, default 3</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">            5</span></div><div>                            ####-d option is to control how to call discordant read pairs. It is equivalent to define what is the largest fragment (median insert size + d x sd) that could be concordant. Use default if the insert size distribution is narrow and symmetric (top 2 plots in next page), and use 5 if it’s wide or skewed with long tail (bottom 2 plots in next page). Don’t use value less than</div><div style="margin-left: 80px;">    -c FLT    standard deviation cutoff to cluster discordant read pairs, default equal to -d, it is recommended to use same -c as -d if -d&lt;=5, -c 5 if -d &gt; 5</div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -p INT    number of supporting read pairs required for an event to be called, default 2</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">            3</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -o INT    number of supporting full length read pairs, default 0, specify this option will decrease sensitivity on small complex events</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">            1</span></div><div style="margin-left: 80px;">   <span style="color: rgb(255, 0, 0);"> -q INT    number of supporting split reads required, default 1</span></div><div style="margin-left: 80px;">    -z INT    event size cutoff, default 1,000,000,000</div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -s INT    bp to be cut off from beginning and end of unmapped reads, must be same as -s in pre_process.pl, default 20</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">            20</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -m INT    [0/1], if set to 1, use Meerkat to remove duplicates; if set to 0, use flag 'd' marked by Picard or other tools to remove duplicates.</span> <span style="color: rgb(255, 0, 0); font-weight: bold;">If the bam file is aligned by bwa mem, it has to be processed by Picard to mark duplicates and use 0 option.</span> <span style="color: rgb(255, 0, 0);">The bwa mem aligned bam file won't work with option 1. Default 1</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">            0</span></div><div style="margin-left: 80px;">    -a INT    [0/1], adjust non-uniq mapped reads, default 1</div><div style="margin-left: 80px;">    -u INT  [0/1], use all alignments in the BAM file, turn this option on if the BAM file is not generated by BWA, turn on this option will force turning off option a, default 0</div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -Q INT    minimum mapping quality for reads to be used, default 0</span></div><div style="margin-left: 80px;">                10</div><div style="margin-left: 80px;">    -g INT    number of alternative mappings to consider in main bam file, number of alternative mappings printed out in XA tag by bwa is controlled by -N, default use all in bam file</div><div style="margin-left: 80px;">    -f INT    number of alternative mappings to consider in clipped alignments, default use all in bam file</div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -l INT    [0/1], consider clipped alignments, default 1</span></div><div style="margin-left: 80px;">                0</div><div style="margin-left: 80px;">    -t INT    number of threads used in bwa alignment, default 1</div><div style="margin-left: 80px;">    -R STR    file name of read group to be ignored, one read group ID per line</div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -F STR    /path/to/reference/fasta/files, path only, not the files, required</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -S STR    /path/to/samtools, path only, not the command, no need to specify if samtools is in PATH</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -W STR    /path/to/bwa, path only, not the command, no need to specify if bwa is in PATH</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">    -B STR    /path/to/blastall and formatdb, path only, not the command, no need to specify if blastall and formatdb is in PATH</span></div><div style="margin-left: 80px;"><span style="color: rgb(255, 70, 53);">    -P STR    specify step to run, dc|cl|mpd|alg|srd|rf|all, default 'all', each step require results from previous steps</span></div><div style="margin-left: 80px;">        dc: extract discordant read pairs</div><div style="margin-left: 80px;">        cl: construct clusters of discordant read pairs</div><div style="margin-left: 80px;">        mpd: call events based on read pairs</div><div style="margin-left: 80px;">        alg: align split reads to candidate break point regions</div><div style="margin-left: 80px;">        srd: confirm events based on split reads and filter results</div><div style="margin-left: 80px;">        rf:  refine break points by local alignments</div><div style="margin-left: 80px;"><span style="color: rgb(255, 70, 53);">        all: run all above steps</span></div><div>                    ####For example, </div><div style="margin-left: 80px;">for 50bp reads, &lt;10x TCGA genomes, suggest to use “-s 18 -d 5 -a 0 -l 0 -q 1”.</div><div style="margin-left: 80px;">For 75-101bp reads, 30-40x TCGA genomes, suggest to use “-s 20 -d 5 -p 3 -o 1 -a 0 -u 1 -Q 10” (“-a 0 -u 1 -Q 10” is used because most of these genomes lack XT and XA tags so the -a 1 option won’t work). </div><div style="margin-left: 80px;"><span style="color: rgb(255, 0, 0);">If aligned by BWA mem, use “-s 20 -d 5 -p 3 -o 1 -m 0 -l 0”. </span></div><div style="margin-left: 80px;"> For 101bp reads, 60-80x TCGA genomes, suggest to use “-s 20 -d 5 -p 5 -o 3”. </div><div style="margin-left: 80px;">If tumor genome is sequenced at 60x and normal genome at 30x, use 60x parameters for both tumor and normal. Always replace the *.blacklist.gz file</div><div style="margin-left: 80px;">for cancer genome by the file generated from the matched normal genome (discard the cancer.blacklist.gz file, put the normal.blacklist.gz file into the cancer folder and rename it to</div><div style="margin-left: 80px;">cancer.blacklist.gz). For NA18507 run, use “-s 20 -p 3 -o 1”. </div><div style="margin-left: 80px;">For TCGA WES, use same parameter as 30-40x whole genome.</div><div style="margin-left: 80px;"><br/></div><div style="margin-left: 80px;"><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">3）. mechanism.pl</span></div><div style="margin-left: 80px;">mechanism.pl [options]</div><div style="margin-left: 80px;">    -b FILE    sorted and indexed bam file, required</div><div style="margin-left: 80px;">    -o INT    [0/1], include rmsk type &quot;Other&quot; in TE, default 1</div><div style="margin-left: 80px;">    -t INT    max size of TE, default 100,000</div><div style="margin-left: 80px;">    -z INT    size limit of SVs to be processed, default 1,000,000,000</div><div style="margin-left: 80px;">     -R STR    /path/to/repeat_mask_file, required, can be downloaded from UCSC</div><div style="margin-left: 80px;">####<span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">Example：</span></div><div style="margin-left: 80px;">    <span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">perl ./scripts/mechanism.pl -R /db/hg18/rmsk-hg18.txt -b na18507.sorted.bam</span></div><div style="margin-left: 80px;"><font style="font-size: 12pt;"><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif; font-weight: bold;">4）. </span><span style="min-height: 26pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">Outputs</span></font></div><div style="margin-left: 80px;">prefix/*_1/2.fq.gz: artificial read pairs per read group.</div><div style="margin-left: 80px;">prefix/*.is: sample insert size per read group (used to generate prefix.pdf).</div><div style="margin-left: 80px;">prefix.sr/*.srout: split read mapping of each breakpoint, file name is discordant cluster ID. The top line is reference seq on one side of the breakpoint, the bottom line is reference seq on the other side of the breakpoint. The lines in between are reads spanning the breakpoint junction.</div><div style="margin-left: 80px;">prefix.blacklist.gz: positions that shall not be processed.</div><div style="margin-left: 80px;">prefix.bp.fasta: sequences of confidence intervals of break points.</div><div style="margin-left: 80px;">prefix.bp.info: table of cluster id and corresponding break points.</div><div style="margin-left: 80px;">prefix.bp_reads: table of break points and supporting read names.</div><div style="margin-left: 80px;">prefix.cl.disc.sorted.bam: discordant reads in re-mapped artificial read pairs.</div><div style="margin-left: 80px;">prefix.cl.dup.bam: duplicates in re-mapped artificial read pairs.</div><div style="margin-left: 80px;">prefix.cl.sorted.bam: BAM file for alignments of re-mapped artificial read pairs.</div><div style="margin-left: 80px;">prefix.clusters: all discordant clusters sorted by weight (one discordant read pair per line).</div><div style="margin-left: 80px;">prefix.disc.sorted.bam: discordant reads in original BAM file.</div><div style="margin-left: 80px;">prefix.discord: summary of discordant clusters (one discordant cluster per line).</div><div style="margin-left: 80px;">prefix.dre.log: log file to extract discordant read pairs.</div><div style="margin-left: 80px;">prefix.dup.bam: duplicates in original BAM file.</div><div style="margin-left: 80px;">prefix.inter/intra.refined.typ.sorted: variants after refine break points by local alignments, sorted by event type.</div><div style="margin-left: 80px;">prefix.isinfo: read length and insert size statistics per read group.</div><div style="margin-left: 80px;">prefix.mapping.raw: clusters with all possible mapping positions, sorted by coordinates.</div><div style="margin-left: 80px;">prefix.mp.inter/intra.out: variants predicted by discordant read pair clusters.Page 9/20</div><div style="margin-left: 80px;">prefix.pdf: plots of insert size distribution and quality of reads.</div><div style="margin-left: 80px;">prefix.pre.log: log file for extracting unmapped, partially mapped reads and generating artificial read pairs and split reads.</div><div style="margin-left: 80px;">prefix.softclips.fq.gz: partially mapped reads after trimming poor quality basepairs from end.</div><div style="margin-left: 80px;">prefix.softclips.rdist: base quality distribution of partially mapped reads (used to generate prefix.pdf).</div><div style="margin-left: 80px;">prefix.sr.1/2.fq.gz: split read sequences in fastq format.</div><div style="margin-left: 80px;">prefix.sr.inter/intra.filtered: filtered variants of prefix.sr.inter/intra.out (remove variants that use same cluster).</div><div style="margin-left: 80px;">prefix.sr.inter/intra.out: variants have split reads support.</div><div style="margin-left: 80px;">prefix.sr.sorted.bam: split reads alignments.</div><div style="margin-left: 80px;">prefix.unmapped.fq.gz: unmapped reads.</div><div style="margin-left: 80px;">prefix.unmapped.rdist: base quality distribution of unmapped reads (used to generate prefix.pdf).</div><div style="margin-left: 80px;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-weight: bold;">prefix.variants: final variants with mechanism prediction. This is the result file you should look at.</span></div><div style="margin-left: 80px;"><font style="font-size: 12pt;"><span style="min-height: 26pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">5)</span></font><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">Event types:</span></div><div style="margin-left: 80px;">del: simple deletion, no insertion at break point.</div><div style="margin-left: 80px;">del_ins: deletion with insertion at the break point with unknown source.</div><div style="margin-left: 80px;">del_inssd: deletion with insertion at the break point, insertion comes from the same chromosome, same orientation and downstream of deletion.</div><div style="margin-left: 80px;">del_inssu: deletion with insertion at the break point, insertion comes from the same chromosome, same orientation and upstream of deletion.</div><div style="margin-left: 80px;">del_insod: deletion with insertion at the break point, insertion comes from the same chromosome, opposite orientation and downstream of deletion.</div><div style="margin-left: 80px;">del_insou: deletion with insertion at the break point, insertion comes from the same chromosome, opposite orientation and upstream of deletion.</div><div style="margin-left: 80px;">del_inss: deletion with insertion at the break point, insertion comes from a different chromosome, same orientation.</div><div style="margin-left: 80px;">del_inso: deletion with insertion at the break point, insertion comes from a different chromosome, opposite orientation.</div><div style="margin-left: 80px;">del_invers: deletion with inversion at the break point, inversion comes from deleted part.</div><div style="margin-left: 80px;">inssd: insertion, insertion comes from the same chromosome, same orientation and downstream of deletion.</div><div style="margin-left: 80px;">inssu: insertion, insertion comes from the same chromosome, same orientation and upstream of deletion.</div><div style="margin-left: 80px;">insod: insertion, insertion comes from the same chromosome, opposite orientation and downstream of deletion.</div><div style="margin-left: 80px;">insou: insertion, insertion comes from the same chromosome, opposite orientation and upstream of deletion.</div><div style="margin-left: 80px;">inss: insertion, insertion comes from a different chromosome, same orientation.</div><div style="margin-left: 80px;">inso: insertion, insertion comes from a different chromosome, opposite orientation.</div><div style="margin-left: 80px;">invers: inversion with reciprocal discordant read pair cluster support</div><div style="margin-left: 80px;">invers_f: unpaired cluster, both end of read pairs mapped to same chromosome, both on forward strand.</div><div style="margin-left: 80px;">invers_r: unpaired cluster, both end of read pairs mapped to same chromosome, both on reverse strand.</div><div style="margin-left: 80px;">tandem_dup: tandem duplication.</div><div style="margin-left: 80px;">transl_inter: inter chromosomal translocation.</div><div style="margin-left: 80px;"><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">6)Types of mechanisms:</span></font></div><div style="margin-left: 80px;">TEI: transposable element insertion, single or multiple TE insertion.</div><div style="margin-left: 80px;">TEA: Alternative TE, usually a deletion with insertion event is called, deletion is a TE in reference genome and insertion is the same type of TE in reference genome. It should be arisen from sequence divergence of TE.</div><div style="margin-left: 80px;">VNTR: variable number of tandem repeat, deletion or insertion of satellite repeat, simple repeat or low complexity repeat.</div><div style="margin-left: 80px;">NAHR: non-allellic homologous recombination, &gt;100bp homology.</div><div style="margin-left: 80px;">alt-EJ: alternative end joining, 3-100bp homology.</div><div style="margin-left: 80px;">NHEJ: non-homologous end joining, 0-2bp homology or 1-10bp insertion at deletion breakpoint.</div><div style="margin-left: 80px;">FoSTeS: fork stalling and template switching, template switch, &gt;10bp insertion at deletion break points.</div><div style="margin-left: 80px;">NA: unclassified.</div><div style="margin-left: 80px;"><font style="font-size: 12pt;"><span style="font-size: 12pt; font-weight: bold;">7)</span><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">Format for final variants (prefix.variants):</span></font></div><div style="margin-left: 80px;"><span style="font-size: 14pt; font-weight: bold;">    见附件</span></div><div style="margin-left: 40px;"><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">2.4) </span><span style="min-height: 26pt; font-size: 14pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">Filtering Variants(</span></font><span style="font-size: 12pt; min-height: 15pt; font-family: LiberationSerif-Bold; font-weight: bold;">somatic_sv.pl</span><font style="font-size: 14pt;"><span style="min-height: 26pt; font-size: 14pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">):</span></font></div><div><span style="font-size: 10pt;">                        The variants reported in file prefix.variants are raw calls. It should be filtered to produce more reliable calls.</span> <span style="font-size: 10pt; color: rgb(255, 0, 0); font-weight: bold;">If you are calling somatic SVs in cancer genomes, use somatic_sv.pl to filter out germline events and other artifacts.</span> <span style="font-size: 10pt;">It works for both high coverage (&gt;20x) and low coverage genoms (&lt;10x). You can also</span> <span style="font-size: 10pt; color: rgb(255, 0, 0);">use somatic_sv.pl to get high confidence germline SVs in normal genomes (&gt;20x).</span></div><div style="margin-left: 40px;"><span style="font-weight: bold;"> 1)Example of command to get high confidence somatic SVs from either 30-40x or 60x TCGA genomes aligned by either BWA sampe or mem and TCGA WES, all of the following steps need to be run sequentially:</span></div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $variantsfile -o $somatica -F all_normal_disc/ -l 1000 -R /db/hg18/rmsk-hg18.txt</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somatica -o $somaticb -R /db/hg18/rmsk-hg18.txt -n 1 -D 5 -Q 10 -B $normal_bam -I $normal_isinfo -K $normal_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somaticb -o $somaticc -R /db/hg18/rmsk-hg18.txt -u 1 -Q 10 -B $normal_bam -K $normal_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somaticc -o $somaticd -R /db/hg18/rmsk-hg18.txt -f 1 -Q 10 -B $normal_bam -K $normal_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somaticd -o $somatice -R /db/hg18/rmsk-hg18.txt -e 1 -D 5 -Q 10 -B $tumor_bam -I $tumor_isinfo -K $tumor_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somatice -o $somaticf -R /db/hg18/rmsk-hg18.txt -z 1</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -S /home/ly55/opt/samtools/ -i $somaticf -o $somaticg -R /db/hg18/rmsk-hg18.txt -d 40 -t 20</div><div style="margin-left: 40px;"><font color="#FF0000" style="font-size: 12pt;"><br/></font></div><div style="margin-left: 40px;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-weight: bold;">2)Example of command to get high confidence somatic SVs from 6-8x genomes, all of the following steps need to be run sequentially:</span></div><div style="margin-left: 40px;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-weight: bold;">   ####低深度全基因组应该用这一套参数</span></div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $variantsfile -o $somatica -F all_normal_disc/ -R /db/hg19/rmskhg19.txt -x 1</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $somatica -o $somaticb -R /db/hg19/rmsk-hg19.txt -M 1 -B $tumor_bam -C $tumor_bp_reads</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $somatica -o $somaticc -R /db/hg19/rmsk-hg19.txt -L /opt/blat/ -V &quot;10.11.240.76&quot; -T 17777 -A $srout_dir</div><div style="margin-left: 40px;">    perl ./scripts/merge_sv.pl $somaticd $somaticb $somaticc</div><div style="margin-left: 40px;">    We found the above c and d filters are too stringent and will miss some real events, but merging the events together performs better. The $somaticd file is the final output.</div><div style="margin-left: 40px;"><br/></div><div style="margin-left: 40px;"><span style="font-weight: bold;">3)Example of command to get high confidence normal or germline SVs from 30-40x genomes, all of the following steps need to be run sequentially:</span></div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $variantsfile -o $germa -R /db/hg19_bi/rmsk-hg19_bi.txt -E 0</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $germa -o $germb -R /db/hg19_bi/rmsk-hg19_bi.txt -E 0 -e 1 -D 5 -Q 10 -B $normal_bam -I $normal_isinfo -K $normal_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $germb -o $germc -R /db/hg19_bi/rmsk-hg19_bi.txt -E 0 -u 1 -Q 10 -B $normal_bam -K $normal_blacklistrg</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $germc -o $germd -R /db/hg19_bi/rmsk-hg19_bi.txt -E 0 -d 40 -t 40</div><div style="margin-left: 40px;">    perl ./scripts/somatic_sv.pl -i $germd -o $germe -R /db/hg19_bi/rmsk-hg19_bi.txt -E 0 -z 1 -p 5 -P 10</div><div style="margin-left: 40px;">    Eventually, one might consider to manually inspect the discordant read pair alignments with IGV and inspect the split read alignments from prefix.sr folder to make sure the variants are convincing. For split read alignments, one may need to manually edit the alignment a little bit. For the reads spanning the breakpoint junction, part of the reads should align to the left-hand side of reference seq on the top, the remaining of the reads should align to the right-hand side of reference seq on the bottom.</div><div style="margin-left: 40px;"><br/></div><div style="margin-left: 40px;"><span style="font-weight: bold;">4)For TCGA WES, extra filtering steps are needed: 1. discard samples that have &gt;=100 somatic fusions. 2. apply filter_fusions.pl as following:</span></div><div style="margin-left: 40px;">    perl ./scripts/filter_fusions.pl -i $somaticg_fusion -o $somatich_fusion -m 4 -s 20000</div><div style="margin-left: 40px;">    The input file $somaticg_fusion is generated by fusions.pl from the final somatic variants (see section VII.2 for fusions.pl usage).</div><div><br/></div><div>    <span style="font-size: 14pt; font-weight: bold;">2.5) transfomation + annotation + frequency</span><span style="font-size: 14pt; min-height: 26pt; color: rgb(1, 1, 1); font-family: LiberationSerif-Bold; font-weight: bold;">:</span></div><div style="margin-left: 40px;"><span style="font-weight: bold;">1) Convert Meerkat to VCF format (meerkat2vcf.pl):</span></div><div style="margin-left: 40px;">    perl ./scripts/meerkat2vcf.pl -i variantfile -o vcffile -H headerfile -F /db/hg18/hg18_fasta/</div><div style="margin-left: 40px;"><span style="font-weight: bold;">2) Fusion annotation (fusions.pl):</span></div><div style="margin-left: 40px;">    perl ./scripts/fusions.pl -i variantfile -G /db/hg18/refGene_hg18_sorted.txt</div><div style="margin-left: 40px;"><span style="font-weight: bold;">3) Calculate allele frequency (discon.pl):</span></div><div style="margin-left: 40px;">    perl ./scripts/discon.pl -d 5 -Q 10 -i $somaticg -o $somaticg_rp -B $cancer_bam -C $cancer_cluster -I $cancer_isinfo -K $cancer_blacklistrg -S /home/ly55/opt/samtools/</div><div style="margin-left: 80px;">Each event is given an RP tag with 4 numbers in A_B_C_D format.</div><div style="margin-left: 80px;">A: number of full length discordant read pairs</div><div style="margin-left: 80px;">B: number of discordant read pairs from partially mapped reads (clipped reads)</div><div style="margin-left: 80px;">C: number of concordant read pairs at the first breakpoint</div><div style="margin-left: 80px;">D: number of concordant read pairs at the second breakpoint</div><div style="margin-left: 80px;">The number of A+B should be equal to the total number of discordant read pairs given by Meerkat.</div><div>                  <font style="font-size: 12pt;"><span style="font-size: 12pt; color: rgb(255, 0, 0); font-weight: bold;">  AF=(A+B)/(min(C+D)+A+B)</span></font></div><div><br/></div><div><br/></div><div><span style="color: rgb(255, 0, 0);">####</span><span style="min-height: 15pt; font-size: 12pt; color: rgb(255, 0, 0); font-family: LiberationSerif-Bold; font-weight: bold;">Special attention</span></div><div style="min-height: 15pt; text-align: left;"><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">1）When you run Meerkat on cancer genome, it would be better to use the blacklist.gz file </span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">generated from its matched normal genome rather than use the blacklist.gz file generated from the </span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">cancer genome. These files are generated by pre_process.pl.</span></div><div style="min-height: 15pt; text-align: left;"><span style="min-height: 15pt; font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">2）Sometimes, you will get error message &quot;differing read lengths&quot; in pre_process.pl, you can </span><span style="font-size: 12pt; color: rgb(1, 1, 1); font-family: LiberationSerif;">ignore this message. It’s only telling you the read lengths are not the same in some read groups.</span></div><div style="min-height: 15pt; text-align: left;"><span style="min-height: 15pt; font-size: 12pt;"><br/></span></div><div style="min-height: 15pt; text-align: left;"><span style="min-height: 15pt; font-size: 12pt;">####附件</span></div><div><a href="WGS-SV 分析工具之：Meerkat_files/meerkat.zip"><img src="WGS-SV 分析工具之：Meerkat_files/c49fe6167baf76f36a34055a57805650.png" alt="meerkat.zip"></a></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div></span>
</div></body></html> 