<html>
<head>
  <title>第一步--突变注释之VEP：</title>
  <basefont face="微软雅黑" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="YXBJ Windows/600753 (zh-CN, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: 微软雅黑;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="962"/>
<h1>第一步--突变注释之VEP：</h1>

<div>
<span><div><span style="font-size: 16pt; color: rgb(227, 0, 0);">第一步--突变注释之VEP：</span></div><div><span style="font-size: 16pt;"><br/></span></div><div><span style="font-size: 16px; color: rgb(47, 47, 47); font-family: -apple-system, &quot;SF UI Text&quot;, Arial, &quot;PingFang SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei&quot;, &quot;WenQuanYi Micro Hei&quot;, sans-serif;">VEP是国际三大数据库之一的</span><span style="font-size: 16px; background-color: rgb(255, 255, 255); color: rgb(47, 47, 47); font-family: -apple-system, &quot;SF UI Text&quot;, Arial, &quot;PingFang SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei&quot;, &quot;WenQuanYi Micro Hei&quot;, sans-serif;"> </span><span style="font-size: 16px; background-color: rgb(255, 255, 255); color: rgb(47, 47, 47); font-family: -apple-system, &quot;SF UI Text&quot;, Arial, &quot;PingFang SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei&quot;, &quot;WenQuanYi Micro Hei&quot;, sans-serif; font-weight: bold;">ENSEMBL</span><span style="font-size: 16px; background-color: rgb(255, 255, 255); color: rgb(47, 47, 47); font-family: -apple-system, &quot;SF UI Text&quot;, Arial, &quot;PingFang SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei&quot;, &quot;WenQuanYi Micro Hei&quot;, sans-serif;"> </span><span style="font-size: 16px; color: rgb(47, 47, 47); font-family: -apple-system, &quot;SF UI Text&quot;, Arial, &quot;PingFang SC&quot;, &quot;Hiragino Sans GB&quot;, &quot;Microsoft YaHei&quot;, &quot;WenQuanYi Micro Hei&quot;, sans-serif;">提供的，也是非常主流和方便，但它是基于perl语言的。</span></div><div>官网下载地址：<a href="http://asia.ensembl.org/info/docs/tools/vep/script/vep_download.html">http://asia.ensembl.org/info/docs/tools/vep/script/vep_download.html</a></div><div>git下载地址：git clone https://github.com/Ensembl/ensembl-vep.git（cd ensembl-vep，perl INSTALL.pl）</div><div><br/></div><div>1）依赖包（Perl packages）:</div><div>        Archive::Zip        perl -e 'use Archive::Extract'</div><div>        DBD::mysql        perl -e 'use DBD::mysql'</div><div>        DBI</div><div>        </div><div>2）安装：perl INSTALL.pl （运行时注意参数--CACHEDIR [dir]，指定注释数据库地址）</div><div>     docker地址：<a href="https://hub.docker.com/r/ensemblorg/ensembl-vep/">https://hub.docker.com/r/ensemblorg/ensembl-vep/</a></div><div>                           <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep">https://hub.docker.com/r/ensemblorg/ensembl-vep</a></div><div>     docker版本： docker pull ensemblorg/ensembl-vep   (Grch.38)</div><div>                           docker run -t -i ensemblorg/ensembl-vep ./vep</div><div>     docker应用说明：<a href="http://asia.ensembl.org/info/docs/tools/vep/script/vep_download.html">http://asia.ensembl.org/info/docs/tools/vep/script/vep_download.html</a></div><div>     cache下载地址：<a href="https://davetang.org/wiki2/index.php?title=VEP">https://dshiavetang.org/wiki2/index.php?title=VEP</a></div><div><span style="font-size: 12pt; color: rgb(227, 0, 0);">    应用示例：./vep --input_file 199003853TD_199003855TD_chr1_shuf.vcf  --format vcf --output_file 199003853TD_199003855TD_chr1_shuf.vcf.out --cache --offline --force_overwrite  --stats_text  --fork 4 --variant_class --sift b --numbers --transcript_version  --protein --symbol --vcf</span></div><div>./vep \</div><div>--input_file $1 \</div><div>--output_file $2 \</div><div>--format vcf --vcf  \</div><div>--symbol --tsl --terms SO --hgvs --offline  \</div><div>--fasta /mnt/X500/farmers/linhx/bin/ref/hs37d5.fa \</div><div>--cache /mnt/X500/farmers/linhx/bin/software/pvactools/vep_cache/96_GRCh37 \</div><div>--plugin Downstream \</div><div>--plugin Wildtype \</div><div>--dir_plugins /mnt/X500/farmers/linhx/bin/software/pvactools/VEP_plugins \</div><div>--pick  --transcript_version --force_overwrite    \</div><div>--max_af --af_gnomad    （注释频率信息）</div><div><br/></div><div><br/></div><div>3）参数说明官网：<a href="http://asia.ensembl.org/info/docs/tools/vep/script/vep_options.html">http://asia.ensembl.org/info/docs/tools/vep/script/vep_options.html</a></div><div>    参数说明：</div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Basic options</span></div><div>        --species [species]   default = &quot;homo_sapiens&quot;</div><div>        --config [filename]  以config文件读入参数，如 output_file my_output.txt</div><div>        --everything             下面所有参数的集合简写:--sift b, --polyphen b, --ccds, --uniprot, --hgvs, --symbol, --numbers, --domains, </div><div>                                                                                --regulatory, --canonical, --protein, --biotype, --uniprot等等。</div><div><span style="color: rgb(227, 0, 0);">        --input_file [filename]        文件输入</span></div><div>        --input_data [string]        字符串输入，如--input_data rs699</div><div><span style="color: rgb(255, 0, 0);">        --format [format]            输入格式：one of &quot;ensembl&quot;, &quot;vcf&quot;, &quot;hgvs&quot;, &quot;id&quot;, &quot;region&quot;, &quot;spdi&quot;.</span></div><div><span style="color: rgb(227, 0, 0);">        --output_file [filename]    输出文件</span></div><div><span style="color: rgb(227, 0, 0);">        --force_overwrite            强制覆盖输出文件</span></div><div><span style="color: rgb(255, 0, 0);">        --stats_file [filename]        如stat.html，注释结果统计文件</span></div><div>        --stats_text                统计结果输出为txt</div><div>        --no_stats                    不需要统计</div><div>        --warning_file [filename]    注释错误和警告输出文件</div><div><span style="color: rgb(227, 0, 0);">        --max_sv_size                最大SV检测</span></div><div>        --no_check_variants_order    允许输入文件未经排序，但是更加耗时，耗内存</div><div><span style="color: rgb(227, 0, 0);">        --fork [num_forks]            可以理解为线程数</span></div><div><br/></div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Cache options</span></div><div><span style="color: rgb(227, 0, 0);">        --cache        Enables use of the cache. Add --refseq or --merged to use the refseq or merged cache, (if installed).</span></div><div><span style="color: rgb(227, 0, 0);">        --dir [directory]    Specify the base cache/plugin directory to use. Default = &quot;$HOME/.vep/&quot;</span></div><div>        --dir_cache [directory]        Specify the cache directory to use. Default = &quot;$HOME/.vep/&quot;</div><div>        --dir_plugins [directory]    Specify the plugin directory to use. Default = &quot;$HOME/.vep/&quot;</div><div><span style="color: rgb(227, 0, 0);">        --offline            Enable offline mode. No database connections will be made, and a cache file or GFF/GTF file is required for annotation</span></div><div><span style="color: rgb(227, 0, 0);">        --fasta [file|dir]    Specify a FASTA file or a directory containing FASTA files</span></div><div><span style="color: rgb(227, 0, 0);">        --refseq    Specify this option if you have installed the RefSeq cache in order for VEP to pick up the alternate cache directory.</span></div><div>        --merged    Use the merged Ensembl and RefSeq cache. Consequences are flagged with the SOURCE of each transcript used.</div><div>        --show_cache_info    Show source version information for selected cache and quit</div><div><span style="color: rgb(227, 0, 0);">        --buffer_size    指定内存大小</span></div><div>        </div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Other annotation sources</span></div><div><span style="color: rgb(227, 0, 0);">        --gff [filename]        Use GFF transcript annotations in [filename] as an annotation source. Requires a FASTA file of genomic sequence.Not used by default</span></div><div><span style="color: rgb(227, 0, 0);">        --gtf [filename]        Use GTF transcript annotations in [filename] as an annotation source. Requires a FASTA file of genomic sequence.Not used by default</span></div><div>        </div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Output format options</span></div><div><span style="color: rgb(227, 0, 0);">        --vcf            Writes output in VCF format</span></div><div>        --tab            Writes output in tab-delimited format. Not used by default</div><div>        --json            Writes output in JSON format. Not used by default</div><div>        --compress_output [gzip|bgzip]    Writes output compressed using either gzip or bgzip. Not used by default</div><div>        --fields [list]        Configure the output format using a comma separated list of fields.Can only be used with tab (--tab) or VCF format (--vcf) output.</div><div>        </div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Output options</span></div><div><span style="color: rgb(227, 0, 0);">        --variant_class        Output the Sequence Ontology variant class. （snv,indel,copy_number_gain...）</span></div><div><span style="color: rgb(227, 0, 0);">        --sift [p|s|b]        VEP can output the prediction term, score or both</span></div><div><span style="color: rgb(227, 0, 0);">        -polyphen [p|s|b]    VEP can output the prediction term, score or both</span></div><div>        --humdiv            Human only Retrieve the humDiv PolyPhen prediction instead of the default humVar.</div><div>        --nearest [transcript|gene|symbol]        Retrieve the transcript or gene with the nearest protein-coding transcription start site (TSS) to each input variant.</div><div>        --distance [bp_distance(,downstream_distance_if_different)]   Modify the distance up and/or downstream between a variant and a transcript</div><div>        --overlaps        Report the proportion and length of a transcript overlapped by a structural variant in VCF format.</div><div>        --gene_phenotype        Indicates if the overlapped gene is associated with a phenotype, disease or trait. See list of phenotype sources. Not used by default</div><div><span style="color: rgb(227, 0, 0);">        --cell_type        Report only regulatory regions that are found in the given cell type(s).</span>     </div><div><span style="color: rgb(227, 0, 0);">        --total_length        Give cDNA, CDS and protein positions as Position/Length. Not used by default</span></div><div><span style="color: rgb(227, 0, 0);">        --numbers        Adds affected exon and intron numbering to to output. Format is Number/Total. Not used by default</span></div><div><span style="font-size: 14pt; color: rgb(255, 0, 0);">        Identifiers</span></div><div>        --hgvs        Add HGVS nomenclature based on Ensembl stable identifiers to the output.</div><div>        --hgvsg        Add genomic HGVS nomenclature based on the input chromosome name.</div><div><span style="color: rgb(255, 0, 0);">        --transcript_version        Add version numbers to Ensembl transcript identifiers</span></div><div><span style="color: rgb(255, 0, 0);">        --protein        Add the Ensembl protein identifier to the output where appropriate. Not used by default</span></div><div><span style="color: rgb(255, 0, 0);">        --symbol        dds the gene symbol (e.g. HGNC) (where available) to the output. Not used by default</span></div><div><span style="color: rgb(255, 0, 0);">        --ccds        Adds the CCDS transcript identifer (where available) to the output. Not used by default</span></div><div>        --uniprot        Adds best match accessions for translated protein products from three UniProt-related databases (SWISSPROT, TREMBL and UniParc) to the output. Not used by     default</div><div><span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;;">        </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal; line-height: 16px;">--tsl    </span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">Adds the</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><a href="http://asia.ensembl.org/Help/Glossary?id=492" style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">transcript support level</a><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;">for this transcript to the output.</span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-style: italic; font-variant-caps: normal; font-variant-ligatures: normal;">Not used by default (</span><a href="http://asia.ensembl.org/Help/Glossary?id=492" style="font-size: 10pt; color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-style: italic;">http://asia.ensembl.org/Help/Glossary?id=492</a><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-style: italic; font-variant-caps: normal; font-variant-ligatures: normal;">)</span></div><div><font face="Times New Roman" style="font-size: 10pt;"><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; font-family: &quot;Times New Roman&quot;; font-style: italic; font-variant-caps: normal; font-variant-ligatures: normal;">                    </span></font> <span style="font-size: 12.8px; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;, Helvetica, Arial, Geneva, sans-serif; font-variant-caps: normal; font-variant-ligatures: normal;">The Transcript Support Level (TSL) is a method to highlight the well-supported and poorly-supported transcript models for users, based on the type and quality of the alignments used to annotate the transcript.</span></div><div><span style="font-size: 9pt;"><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;; font-style: italic;">                    tsl1 – all splice junctions of the transcript are supported by at least one non-suspect mRNA</span></span></div><div><span style="font-size: 9pt;"><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;; font-style: italic;">                    tsl2 – the best supporting mRNA is flagged as suspect or the support is from multiple ESTs</span></span></div><div><span style="font-size: 9pt;"><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;; font-style: italic;">                    tsl3 – the only support is from a single EST</span></span></div><div><span style="font-size: 9pt;"><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;; font-style: italic;">                    tsl4 – the best supporting EST is flagged as suspect</span></span></div><div><span style="font-size: 9pt;"><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: &quot;Luxi Sans&quot;; font-style: italic;">                    tsl5 – no single transcript supports the model structure</span></span></div><div><span style="font-size: 9pt;"><br/></span></div><div> <span style="font-size: 14pt; color: rgb(255, 0, 0);">   </span><span style="font-size: 14pt; color: rgb(255, 0, 0);">    Co-located variants</span></div><div>        --af        Add the global allele frequency (AF) from 1000 Genomes Phase 3 data for any known co-located variant to the output.</div><div>        --max_af        Report the highest allele frequency observed in any population from 1000 genomes, ESP or gnomAD. Not used by default</div><div>        --af_1kg            Add allele frequency from continental populations (AFR,AMR,EAS,EUR,SAS) of 1000 Genomes Phase 3 to the output. Must be used with --cache. Not used by default</div><div>        --af_esp            Include allele frequency from NHLBI-ESP populations. Must be used with --cache. Not used by default</div><div>        --af_gnomad            Include allele frequency from Genome Aggregation Database (gnomAD) exome populations.</div><div>        --af_exac            Include allele frequency from ExAC project populations. Must be used with --cache. Not used by default</div><div>        --pubmed            Report Pubmed IDs for publications that cite existing variant. Must be used with --cache. Not used by default</div><div>        --failed [0|1]            When checking for co-located variants, by default VEP will exclude variants that have been flagged as failed. Set this flag to include such variants. Default: 0 (exclude)</div><div><span style="font-size: 14pt; color: rgb(227, 0, 0);">        Filtering and QC options</span></div><div>        --gencode_basic        Limit your analysis to transcripts belonging to the GENCODE basic set. This set has fragmented or problematic transcripts removed. Not used by default</div><div>        --exclude_predicted        When using the RefSeq or merged cache, exclude predicted transcripts (i.e. those with identifiers beginning with &quot;XM_&quot; or &quot;XR_&quot;).</div><div>        --transcript_filter            ADVANCED Filter transcripts according to any arbitrary set of rules.</div><div>        --check_ref        Force VEP to check the supplied reference allele against the sequence stored in the Ensembl Core database or supplied FASTA file. Lines that do not match are skipped. Not used by default</div><div>        --lookup_ref        Force overwrite the supplied reference allele with the sequence stored in the Ensembl Core database or supplied FASTA file. Not used by default</div><div>        --chr [list]            Select a subset of chromosomes to analyse from your file. Any data not on this chromosome in the input will be skipped. ( --chr 1-3,10,X )</div><div>        --coding_only        Only return consequences that fall in the coding regions of transcripts. Not used by default</div><div>        --no_intergenic            Do not include intergenic consequences in the output. Not used by default</div><div>        --summary        Output only a comma-separated list of all observed consequences per variant. Transcript-specific columns will be left blank. Not used by default    </div><div>        --freq_pop [pop]        Name of the population to use in frequency filter. This must be one of the following:</div><div>        --freq_freq [freq]            Allele frequency to use for filtering. Must be a float value between 0 and 1</div><div>        --freq_gt_lt [gt|lt]        Specify whether the frequency of the co-located variant must be greater than (gt) or less than (lt) the value specified with --freq_freq</div><div>        --freq_filter [exclude|include]        Specify whether to exclude or include only variants that pass the frequency filter</div><div><br/></div><div>        </div></span>
</div></body></html> 